// shadcn-glass-ui/dist/index.mjs
import { t as cn } from "./utils-B792GPM_.mjs";
import { $ as sparklineBarVariants, $n as DropdownMenuSub, $t as Dialog, A as RepositoryCardActions, An as DropdownMenuGlass, Ar as AlertGlass, At as ToggleGlass, B as RepositoryCardRoot, Bn as DropdownMenuGlassShortcut, Bt as skeletonVariants, C as useYearCard, Cn as InputGlass, Cr as AvatarGlassImage, Ct as TooltipGlassContent, D as TrustScoreDisplayGlass, Dn as DropdownMenu, Dr as statusSizes, Dt as TooltipProvider, E as UserInfoGlass, En as DropdownGlass, Er as avatarSizes, Et as TooltipGlassTrigger, F as RepositoryCardLanguages, Fn as DropdownMenuGlassLabel, Fr as FormFieldWrapper, Ft as TabsList, G as useRepositoryCardOptional, Gn as DropdownMenuGroup, Gt as HoverCardGlassTrigger, H as RepositoryCardStatsLine, Hn as DropdownMenuGlassSubContent, Ht as HoverCardGlassContent, I as RepositoryCardMeta, In as DropdownMenuGlassPortal, Ir as TouchTarget, It as TabsTrigger, J as ContributionMetricsGlass, Jn as DropdownMenuPortal, Jt as PopoverGlassContent, K as MetricsGridGlass, Kn as DropdownMenuItem, Kt as PopoverGlass, L as RepositoryCardMetricItem, Ln as DropdownMenuGlassRadioGroup, Lr as ICON_SIZES, Lt as SliderGlass, M as RepositoryCardGlass, Mn as DropdownMenuGlassContent, Mr as AlertGlassTitle, Mt as Tabs, N as RepositoryCardHeader, Nn as DropdownMenuGlassGroup, Nr as AlertTitle, Nt as TabsContent, O as RepositoryMetadataGlass, On as DropdownMenuCheckboxItem, Or as Alert, Ot as TooltipTrigger, P as RepositoryCardIssues, Pn as DropdownMenuGlassItem, Pr as InteractiveCard, Pt as TabsGlass, Q as SparklineGlass, Qn as DropdownMenuShortcut, Qt as notificationVariants, R as RepositoryCardMetrics, Rn as DropdownMenuGlassRadioItem, Rr as alertVariants, Rt as Skeleton, S as YearCardYear, Sn as Input, Sr as AvatarGlassFallback, St as TooltipGlass, T as UserStatsLineGlass, Tn as cardIntensity, Tr as AvatarImage, Tt as TooltipGlassSimple, U as RepositoryCardStatus, Un as DropdownMenuGlassSubTrigger, Ut as HoverCardGlassLegacy, V as RepositoryCardStars, Vn as DropdownMenuGlassSub, Vt as HoverCardGlass, W as useRepositoryCard, Wn as DropdownMenuGlassTrigger, Wt as HoverCardGlassRoot, X as CareerStatsHeaderGlass, Xn as DropdownMenuRadioItem, Xt as PopoverGlassTrigger, Y as CircularMetricGlass, Yn as DropdownMenuRadioGroup, Yt as PopoverGlassLegacy, Z as AICardGlass, Zn as DropdownMenuSeparator, Zt as NotificationGlass, _ as YearCardRoot, _n as ModalPortal, _r as BadgeGlass, _t as insightVariantConfig, a as HeaderNavGlass, an as DialogOverlay, ar as dropdownLabelClasses, at as progressSizes, b as YearCardStats, bn as ModalTrigger, br as AvatarFallback, bt as Tooltip, c as CareerStatsGlass, cn as DialogTrigger, cr as getDropdownItemClasses, ct as FlagAlertGlass, d as YearCardExpandedContent, dn as ModalContent, dr as Checkbox, dt as ThemeToggleGlass, en as DialogClose, er as DropdownMenuSubContent, et as sparklineContainerVariants, f as YearCardGlass, fn as ModalDescription, fr as CheckboxGlass, ft as StatItemGlass, g as YearCardProgress, gn as ModalOverlay, gr as Badge, gt as insightCardVariants, h as YearCardInsights, hn as ModalHeader, hr as buttonGlassVariants, ht as InsightCardGlass, i as ProfileHeaderExtendedGlass, in as DialogHeader, ir as dropdownContentClasses, it as ProgressGlass, j as RepositoryCardExpandedContent, jn as DropdownMenuGlassCheckboxItem, jr as AlertGlassDescription, jt as toggleSizes, k as RepositoryHeaderGlass, kn as DropdownMenuContent, kr as AlertDescription, kt as Toggle, l as YearCardAction, ln as ModalBody, lr as inputVariants, lt as StatusIndicatorGlass, m as YearCardInsightItem, mn as ModalGlass, mr as ButtonGlass, mt as SearchBoxGlass, n as ProjectsListGlass, nn as DialogDescription, nr as DropdownMenuTrigger, nt as RainbowProgressGlass, o as HeaderBrandingGlass, on as DialogPortal, or as dropdownSeparatorClasses, ot as ProfileAvatarGlass, p as YearCardHeader, pn as ModalFooter, pr as Button, pt as SortDropdownGlass, q as MetricCardGlass, qn as DropdownMenuLabel, qt as PopoverGlassAnchor, r as ProfileHeaderGlass, rn as DialogFooter, rr as ComboBoxGlass, rt as Progress, s as FlagsSectionGlass, sn as DialogTitle, sr as getDropdownContentStyles, st as LanguageBarGlass, t as TrustScoreCardGlass, tn as DialogContent, tr as DropdownMenuSubTrigger, tt as SegmentedControlGlass, u as YearCardBadge, un as ModalClose, ur as CircularProgressGlass, ut as BaseProgressGlass, v as YearCardSparkline, vn as ModalRoot, vr as badgeVariants, vt as IconButtonGlass, w as useYearCardOptional, wn as GlassCard, wr as AvatarGlassSimple, wt as TooltipGlassProvider, x as YearCardValue, xn as modalSizes, xr as AvatarGlass, xt as TooltipContent, y as YearCardStatItem, yn as ModalTitle, yr as Avatar, yt as ExpandableHeaderGlass, z as RepositoryCardName, zn as DropdownMenuGlassSeparator, zt as SkeletonGlass } from "./trust-score-card-glass-Wpva6x8c.mjs";
import { n as useHover, t as useFocus } from "./use-focus-C5kPAKr_.mjs";
import { a as getThemeConfig, i as getNextTheme, n as THEME_CONFIG, o as useTheme, r as ThemeProvider, t as THEMES } from "./theme-context-DLS2uAgJ.mjs";
import { n as useResponsive, t as useWallpaperTint } from "./use-wallpaper-tint-C0kYXNiN.mjs";
import * as React from "react";
import { createContext, forwardRef, useCallback, useContext, useEffect, useId, useLayoutEffect, useMemo, useRef, useState } from "react";
import { Check, ChevronDown, ChevronUp, Menu, PanelLeft, PanelLeftClose, PanelLeftOpen, X } from "lucide-react";
import { cva } from "class-variance-authority";
import { Fragment, jsx, jsxs } from "react/jsx-runtime";
import { Slot } from "@radix-ui/react-slot";
import * as DialogPrimitive from "@radix-ui/react-dialog";
import * as SeparatorPrimitive from "@radix-ui/react-separator";
import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area";
import * as SwitchPrimitive from "@radix-ui/react-switch";
import * as SelectPrimitive from "@radix-ui/react-select";
import * as RechartsPrimitive from "recharts";
import { Bar, BarChart, Cell, ResponsiveContainer } from "recharts";
var blurMap = {
	subtle: "var(--blur-sm)",
	medium: "var(--blur-md)",
	strong: "var(--blur-lg)"
};
var bgVarMap = {
	subtle: "var(--card-subtle-bg)",
	medium: "var(--card-medium-bg)",
	strong: "var(--card-strong-bg)"
};
var borderVarMap = {
	subtle: "var(--card-subtle-border)",
	medium: "var(--card-medium-border)",
	strong: "var(--card-strong-border)"
};
var glowVarMap = {
	blue: "var(--glow-blue)",
	violet: "var(--glow-violet)",
	cyan: "var(--glow-cyan)"
};
function CardGlassRoot({ className, intensity = "medium", glow = null, hover = false, style, ...props }) {
	const cardStyles = {
		background: bgVarMap[intensity],
		borderColor: borderVarMap[intensity],
		backdropFilter: `blur(${blurMap[intensity]})`,
		WebkitBackdropFilter: `blur(${blurMap[intensity]})`,
		boxShadow: glow ? glowVarMap[glow] : "var(--glow-neutral)",
		...style
	};
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card",
		className: cn("flex flex-col gap-6 rounded-xl border py-6 transition-all duration-300", "text-[var(--text-primary)]", hover && "hover:shadow-lg hover:border-[var(--card-hover-border)] cursor-pointer", className),
		style: cardStyles,
		...props
	});
}
function CardGlassHeader({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-header",
		className: cn("@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-2 px-6", "has-data-[slot=card-action]:grid-cols-[1fr_auto]", "[.border-b]:pb-6", className),
		...props
	});
}
function CardGlassTitle({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-title",
		className: cn("leading-none font-semibold text-[var(--text-primary)]", className),
		...props
	});
}
function CardGlassDescription({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-description",
		className: cn("text-sm text-[var(--text-muted)]", className),
		...props
	});
}
function CardGlassAction({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-action",
		className: cn("col-start-2 row-span-2 row-start-1 self-start justify-self-end", className),
		...props
	});
}
function CardGlassContent({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-content",
		className: cn("px-6", className),
		...props
	});
}
function CardGlassFooter({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "card-footer",
		className: cn("flex items-center gap-2 px-6 [.border-t]:pt-6", className),
		...props
	});
}
const CardGlass = {
	Root: CardGlassRoot,
	Header: CardGlassHeader,
	Title: CardGlassTitle,
	Description: CardGlassDescription,
	Action: CardGlassAction,
	Content: CardGlassContent,
	Footer: CardGlassFooter
};
const Card = CardGlassRoot;
const CardHeader = CardGlassHeader;
const CardTitle = CardGlassTitle;
const CardDescription = CardGlassDescription;
const CardContent = CardGlassContent;
const CardFooter = CardGlassFooter;
const sheetVariants = cva([
	"fixed z-50 flex flex-col gap-4 p-6",
	"transition-transform ease-in-out duration-300",
	"data-[state=open]:animate-in data-[state=closed]:animate-out",
	"data-[state=closed]:duration-300 data-[state=open]:duration-500"
].join(" "), {
	variants: { side: {
		top: [
			"inset-x-0 top-0",
			"data-[state=closed]:slide-out-to-top",
			"data-[state=open]:slide-in-from-top",
			"rounded-b-2xl"
		].join(" "),
		bottom: [
			"inset-x-0 bottom-0",
			"data-[state=closed]:slide-out-to-bottom",
			"data-[state=open]:slide-in-from-bottom",
			"rounded-t-2xl"
		].join(" "),
		left: [
			"inset-y-0 left-0 h-full w-3/4 sm:max-w-sm",
			"data-[state=closed]:slide-out-to-left",
			"data-[state=open]:slide-in-from-left",
			"rounded-r-2xl"
		].join(" "),
		right: [
			"inset-y-0 right-0 h-full w-3/4 sm:max-w-sm",
			"data-[state=closed]:slide-out-to-right",
			"data-[state=open]:slide-in-from-right",
			"rounded-l-2xl"
		].join(" ")
	} },
	defaultVariants: { side: "right" }
});
function SheetRoot({ ...props }) {
	return /* @__PURE__ */ jsx(DialogPrimitive.Root, {
		"data-slot": "sheet",
		...props
	});
}
function SheetTrigger({ ...props }) {
	return /* @__PURE__ */ jsx(DialogPrimitive.Trigger, {
		"data-slot": "sheet-trigger",
		...props
	});
}
function SheetPortal({ ...props }) {
	return /* @__PURE__ */ jsx(DialogPrimitive.Portal, {
		"data-slot": "sheet-portal",
		...props
	});
}
var SheetOverlay = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(DialogPrimitive.Overlay, {
	ref,
	"data-slot": "sheet-overlay",
	className: cn("fixed inset-0 z-50", "data-[state=open]:animate-in data-[state=closed]:animate-out", "data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0", className),
	style: {
		background: "var(--modal-overlay)",
		backdropFilter: "blur(var(--blur-sm))",
		WebkitBackdropFilter: "blur(var(--blur-sm))"
	},
	...props
}));
SheetOverlay.displayName = "SheetOverlay";
var SheetContent = React.forwardRef(({ className, children, side = "right", showCloseButton = true, ...props }, ref) => /* @__PURE__ */ jsxs(SheetPortal, { children: [/* @__PURE__ */ jsx(SheetOverlay, {}), /* @__PURE__ */ jsxs(DialogPrimitive.Content, {
	ref,
	"data-slot": "sheet-content",
	"data-side": side,
	className: cn(sheetVariants({ side }), className),
	style: {
		background: "var(--modal-bg)",
		border: "1px solid var(--modal-border)",
		boxShadow: "var(--modal-glow)",
		backdropFilter: "blur(var(--blur-lg))",
		WebkitBackdropFilter: "blur(var(--blur-lg))"
	},
	...props,
	children: [
		/* @__PURE__ */ jsx("div", {
			className: "absolute inset-0 pointer-events-none",
			style: {
				background: "var(--modal-glow-effect)",
				borderRadius: "inherit"
			},
			"aria-hidden": "true"
		}),
		/* @__PURE__ */ jsx("div", {
			className: "relative flex-1 flex flex-col overflow-auto",
			children
		}),
		showCloseButton && /* @__PURE__ */ jsxs(DialogPrimitive.Close, {
			"data-slot": "sheet-close",
			className: cn("absolute top-4 right-4", "p-1.5 md:p-2 rounded-xl", "transition-all duration-300", "ring-offset-background", "focus:outline-none focus:ring-2 focus:ring-(--semantic-primary) focus:ring-offset-2", "hover:opacity-100 opacity-70", "disabled:pointer-events-none"),
			style: {
				background: "var(--modal-close-btn-bg)",
				border: "var(--modal-close-btn-border)",
				color: "var(--text-muted)"
			},
			children: [/* @__PURE__ */ jsx(X, { className: ICON_SIZES.md }), /* @__PURE__ */ jsx("span", {
				className: "sr-only",
				children: "Close"
			})]
		})
	]
})] }));
SheetContent.displayName = "SheetContent";
function SheetHeader({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "sheet-header",
		className: cn("flex flex-col gap-2 text-center sm:text-left", className),
		...props
	});
}
function SheetFooter({ className, ...props }) {
	return /* @__PURE__ */ jsx("div", {
		"data-slot": "sheet-footer",
		className: cn("flex flex-col-reverse gap-2 sm:flex-row sm:justify-end mt-auto pt-4", className),
		...props
	});
}
var SheetTitle = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(DialogPrimitive.Title, {
	ref,
	"data-slot": "sheet-title",
	className: cn("text-lg md:text-xl font-semibold leading-none tracking-tight", className),
	style: { color: "var(--text-primary)" },
	...props
}));
SheetTitle.displayName = "SheetTitle";
var SheetDescription = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(DialogPrimitive.Description, {
	ref,
	"data-slot": "sheet-description",
	className: cn("text-sm", className),
	style: { color: "var(--text-muted)" },
	...props
}));
SheetDescription.displayName = "SheetDescription";
function SheetClose({ ...props }) {
	return /* @__PURE__ */ jsx(DialogPrimitive.Close, {
		"data-slot": "sheet-close",
		...props
	});
}
const SheetGlass = {
	Root: SheetRoot,
	Trigger: SheetTrigger,
	Portal: SheetPortal,
	Overlay: SheetOverlay,
	Content: SheetContent,
	Header: SheetHeader,
	Footer: SheetFooter,
	Title: SheetTitle,
	Description: SheetDescription,
	Close: SheetClose
};
const Sheet = SheetRoot;
function getSeparatorStyles(glow) {
	return {
		background: "var(--separator-bg)",
		...glow && { boxShadow: "var(--separator-glow)" }
	};
}
var SeparatorGlass = React.forwardRef(({ className, orientation = "horizontal", decorative = true, glow = false, ...props }, ref) => /* @__PURE__ */ jsx(SeparatorPrimitive.Root, {
	ref,
	"data-slot": "separator-glass",
	decorative,
	orientation,
	className: cn("shrink-0", "data-[orientation=horizontal]:h-px data-[orientation=horizontal]:w-full", "data-[orientation=vertical]:h-full data-[orientation=vertical]:w-px", "transition-shadow duration-200", className),
	style: getSeparatorStyles(glow),
	...props
}));
SeparatorGlass.displayName = "SeparatorGlass";
function getThumbStyles() {
	return {
		background: "var(--scroll-thumb-bg)",
		border: "1px solid var(--scroll-thumb-border)"
	};
}
var ScrollBarGlass = React.forwardRef(({ className, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsx(ScrollAreaPrimitive.ScrollAreaScrollbar, {
	ref,
	"data-slot": "scroll-area-scrollbar-glass",
	orientation,
	className: cn("flex touch-none select-none transition-colors", orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent p-[2px]", orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent p-[2px]", className),
	...props,
	children: /* @__PURE__ */ jsx(ScrollAreaPrimitive.ScrollAreaThumb, {
		"data-slot": "scroll-area-thumb-glass",
		className: cn("relative flex-1 rounded-full", "transition-colors duration-150", "hover:bg-[var(--scroll-thumb-hover-bg)]"),
		style: getThumbStyles()
	})
}));
ScrollBarGlass.displayName = "ScrollBarGlass";
var ScrollAreaGlass = React.forwardRef(({ className, children, orientation = "vertical", ...props }, ref) => /* @__PURE__ */ jsxs(ScrollAreaPrimitive.Root, {
	ref,
	"data-slot": "scroll-area-glass",
	className: cn("relative overflow-hidden", className),
	...props,
	children: [
		/* @__PURE__ */ jsx(ScrollAreaPrimitive.Viewport, {
			"data-slot": "scroll-area-viewport-glass",
			className: cn("size-full rounded-[inherit]", "focus-visible:ring-[var(--focus-ring)] focus-visible:ring-2 focus-visible:outline-none"),
			children
		}),
		(orientation === "vertical" || orientation === "both") && /* @__PURE__ */ jsx(ScrollBarGlass, { orientation: "vertical" }),
		(orientation === "horizontal" || orientation === "both") && /* @__PURE__ */ jsx(ScrollBarGlass, { orientation: "horizontal" }),
		/* @__PURE__ */ jsx(ScrollAreaPrimitive.Corner, {
			"data-slot": "scroll-area-corner-glass",
			style: { background: "var(--scroll-track-bg)" }
		})
	]
}));
ScrollAreaGlass.displayName = "ScrollAreaGlass";
var ScrollAreaGlassViewport = ScrollAreaPrimitive.Viewport;
ScrollAreaGlassViewport.displayName = "ScrollAreaGlassViewport";
const stepperRootVariants = cva("flex w-full", {
	variants: { orientation: {
		horizontal: "flex-col",
		vertical: "flex-col"
	} },
	defaultVariants: { orientation: "horizontal" }
});
const stepperListVariants = cva("flex", {
	variants: { orientation: {
		horizontal: "flex-row items-center w-full",
		vertical: "flex-col gap-0"
	} },
	defaultVariants: { orientation: "horizontal" }
});
const stepperStepContainerVariants = cva("relative flex transition-all duration-300", {
	variants: { orientation: {
		horizontal: "flex-col items-center flex-1",
		vertical: "flex-row items-start gap-3"
	} },
	defaultVariants: { orientation: "horizontal" }
});
const stepperIndicatorVariants = cva("relative flex items-center justify-center rounded-full font-medium transition-all duration-300 shrink-0", {
	variants: {
		size: {
			sm: "w-8 h-8 text-xs",
			md: "w-10 h-10 text-sm",
			lg: "w-12 h-12 text-base"
		},
		variant: {
			numbered: "",
			icon: "",
			dots: ""
		}
	},
	defaultVariants: {
		size: "md",
		variant: "numbered"
	},
	compoundVariants: [
		{
			variant: "dots",
			size: "sm",
			className: "!w-2.5 !h-2.5"
		},
		{
			variant: "dots",
			size: "md",
			className: "!w-3 !h-3"
		},
		{
			variant: "dots",
			size: "lg",
			className: "!w-4 !h-4"
		}
	]
});
const stepperConnectorVariants = cva("transition-all duration-500 ease-out", {
	variants: { orientation: {
		horizontal: "h-0.5 flex-1 mx-2 md:mx-3",
		vertical: "w-0.5 min-h-6 ml-5 my-1"
	} },
	defaultVariants: { orientation: "horizontal" }
});
const stepperLabelVariants = cva("font-medium transition-colors duration-300 whitespace-nowrap", {
	variants: {
		size: {
			sm: "text-xs",
			md: "text-sm",
			lg: "text-base"
		},
		orientation: {
			horizontal: "mt-2 text-center",
			vertical: ""
		}
	},
	defaultVariants: {
		size: "md",
		orientation: "horizontal"
	}
});
const stepperDescriptionVariants = cva("transition-colors duration-300", {
	variants: {
		size: {
			sm: "text-[10px]",
			md: "text-xs",
			lg: "text-sm"
		},
		orientation: {
			horizontal: "text-center max-w-[100px] md:max-w-[120px]",
			vertical: "max-w-[200px]"
		}
	},
	defaultVariants: {
		size: "md",
		orientation: "horizontal"
	}
});
const stepperContentVariants = cva("animate-in fade-in-0 duration-200", {
	variants: { orientation: {
		horizontal: "mt-6",
		vertical: "mt-4"
	} },
	defaultVariants: { orientation: "horizontal" }
});
var StepperContext = createContext(null);
var useStepperContext = () => {
	const context = useContext(StepperContext);
	if (!context) throw new Error("Stepper compound components must be used within StepperGlass.Root");
	return context;
};
function getStepStatus(stepValue, currentValue, steps, linear, disabled) {
	if (disabled) return "disabled";
	const stepIndex = steps.indexOf(stepValue);
	const currentIndex = steps.indexOf(currentValue);
	if (stepIndex === -1 || currentIndex === -1) return "pending";
	if (stepIndex === currentIndex) return "active";
	if (stepIndex < currentIndex) return "completed";
	if (linear && stepIndex > currentIndex) return "disabled";
	return "pending";
}
var StepperRoot = ({ value, onValueChange, orientation = "horizontal", variant = "numbered", size = "md", linear = false, children, className }) => {
	const [steps, setSteps] = useState([]);
	const registerStep = useCallback((stepValue, index) => {
		setSteps((prev) => {
			if (prev.includes(stepValue)) return prev;
			const newSteps = [...prev];
			newSteps.splice(index, 0, stepValue);
			return newSteps;
		});
	}, []);
	const unregisterStep = useCallback((stepValue) => {
		setSteps((prev) => prev.filter((s) => s !== stepValue));
	}, []);
	const contextValue = useMemo(() => ({
		value,
		onValueChange,
		orientation,
		variant,
		size,
		linear,
		steps,
		registerStep,
		unregisterStep
	}), [
		value,
		onValueChange,
		orientation,
		variant,
		size,
		linear,
		steps,
		registerStep,
		unregisterStep
	]);
	return /* @__PURE__ */ jsx(StepperContext.Provider, {
		value: contextValue,
		children: /* @__PURE__ */ jsx("div", {
			"data-slot": "stepper",
			className: cn(stepperRootVariants({ orientation }), className),
			"aria-label": "Progress steps",
			children
		})
	});
};
var StepperList = forwardRef(({ children, className, ...props }, ref) => {
	const { orientation } = useStepperContext();
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-slot": "stepper-list",
		role: "tablist",
		"aria-orientation": orientation,
		className: cn(stepperListVariants({ orientation }), className),
		...props,
		children
	});
});
StepperList.displayName = "StepperList";
var StepperStep = forwardRef(({ value: stepValue, label, description, icon, completedIcon, disabled: forcedDisabled, className, index: providedIndex }, ref) => {
	const { value: currentValue, onValueChange, orientation, variant, size, linear, steps, registerStep, unregisterStep } = useStepperContext();
	const { isFocusVisible, focusProps } = useFocus({ focusVisible: true });
	const [mountIndex] = useState(() => providedIndex ?? Date.now());
	useLayoutEffect(() => {
		registerStep(stepValue, mountIndex);
		return () => unregisterStep(stepValue);
	}, [
		stepValue,
		mountIndex,
		registerStep,
		unregisterStep
	]);
	const status = getStepStatus(stepValue, currentValue, steps, linear, forcedDisabled);
	const stepIndex = steps.indexOf(stepValue);
	const isLast = stepIndex === steps.length - 1;
	const isClickable = status !== "disabled";
	const indicatorStyles = {
		background: status === "completed" ? "var(--stepper-step-completed-bg)" : status === "active" ? "var(--stepper-step-active-bg)" : status === "disabled" ? "var(--stepper-step-disabled-bg)" : "var(--stepper-step-bg)",
		border: `2px solid ${status === "completed" ? "var(--stepper-step-completed-border)" : status === "active" ? "var(--stepper-step-active-border)" : status === "disabled" ? "var(--stepper-step-disabled-border)" : "var(--stepper-step-border)"}`,
		color: status === "completed" ? "var(--stepper-step-completed-text)" : status === "active" ? "var(--stepper-step-active-text)" : status === "disabled" ? "var(--stepper-step-disabled-text)" : "var(--stepper-step-text)",
		boxShadow: status === "active" ? "var(--stepper-step-active-glow)" : status === "completed" ? "var(--stepper-step-glow)" : isFocusVisible ? "var(--focus-glow)" : "none",
		backdropFilter: "blur(var(--blur-sm))"
	};
	const connectorStyles = { background: stepIndex < steps.indexOf(currentValue) ? "var(--stepper-connector-active-bg)" : "var(--stepper-connector-bg)" };
	const labelStyles = { color: status === "active" || status === "completed" ? "var(--stepper-label-text)" : "var(--stepper-description-text)" };
	const descriptionStyles = { color: "var(--stepper-description-text)" };
	const renderIndicatorContent = () => {
		if (status === "completed") {
			if (completedIcon) return completedIcon;
			return /* @__PURE__ */ jsx(Check, { className: "w-4 h-4" });
		}
		if (variant === "icon" && icon) return icon;
		if (variant === "dots") return null;
		return stepIndex >= 0 ? stepIndex + 1 : "";
	};
	const handleClick = () => {
		if (isClickable && onValueChange) onValueChange(stepValue);
	};
	const handleKeyDown = (e) => {
		if (!isClickable) return;
		const stepList = e.currentTarget.closest("[role=\"tablist\"]");
		if (!stepList) return;
		const allSteps = Array.from(stepList.querySelectorAll("[role=\"tab\"]:not([aria-disabled=\"true\"])"));
		const currentIdx = allSteps.indexOf(e.currentTarget);
		let nextIdx = currentIdx;
		const isHorizontal = orientation === "horizontal";
		switch (e.key) {
			case isHorizontal ? "ArrowRight" : "ArrowDown":
				e.preventDefault();
				nextIdx = (currentIdx + 1) % allSteps.length;
				break;
			case isHorizontal ? "ArrowLeft" : "ArrowUp":
				e.preventDefault();
				nextIdx = currentIdx - 1 < 0 ? allSteps.length - 1 : currentIdx - 1;
				break;
			case "Home":
				e.preventDefault();
				nextIdx = 0;
				break;
			case "End":
				e.preventDefault();
				nextIdx = allSteps.length - 1;
				break;
			default: return;
		}
		const nextStep = allSteps[nextIdx];
		if (nextStep) {
			nextStep.focus();
			const nextValue = nextStep.getAttribute("data-value");
			if (nextValue && onValueChange) onValueChange(nextValue);
		}
	};
	if (orientation === "horizontal") return /* @__PURE__ */ jsxs(Fragment, { children: [/* @__PURE__ */ jsxs("div", {
		"data-slot": "stepper-item",
		className: cn(stepperStepContainerVariants({ orientation }), className),
		children: [/* @__PURE__ */ jsx("button", {
			ref,
			type: "button",
			role: "tab",
			"aria-selected": status === "active",
			"aria-disabled": !isClickable,
			"aria-current": status === "active" ? "step" : void 0,
			"data-value": stepValue,
			"data-status": status,
			onClick: handleClick,
			onKeyDown: handleKeyDown,
			onFocus: focusProps.onFocus,
			onBlur: focusProps.onBlur,
			disabled: !isClickable,
			className: cn(stepperIndicatorVariants({
				size,
				variant
			}), !isClickable && "cursor-not-allowed opacity-60", isClickable && "cursor-pointer hover:scale-105", "min-w-[44px] min-h-[44px]"),
			style: indicatorStyles,
			children: renderIndicatorContent()
		}), label && /* @__PURE__ */ jsxs("div", {
			className: "flex flex-col items-center",
			children: [/* @__PURE__ */ jsx("span", {
				className: cn(stepperLabelVariants({
					size,
					orientation
				})),
				style: labelStyles,
				children: label
			}), description && /* @__PURE__ */ jsx("span", {
				className: cn(stepperDescriptionVariants({
					size,
					orientation
				})),
				style: descriptionStyles,
				children: description
			})]
		})]
	}), !isLast && /* @__PURE__ */ jsx("div", {
		"data-slot": "stepper-separator",
		className: cn(stepperConnectorVariants({ orientation })),
		style: connectorStyles,
		"aria-hidden": "true"
	})] });
	return /* @__PURE__ */ jsxs("div", {
		className: "flex flex-col",
		children: [/* @__PURE__ */ jsxs("div", {
			"data-slot": "stepper-item",
			className: cn(stepperStepContainerVariants({ orientation }), className),
			children: [/* @__PURE__ */ jsx("button", {
				ref,
				type: "button",
				role: "tab",
				"aria-selected": status === "active",
				"aria-disabled": !isClickable,
				"aria-current": status === "active" ? "step" : void 0,
				"data-value": stepValue,
				"data-status": status,
				onClick: handleClick,
				onKeyDown: handleKeyDown,
				onFocus: focusProps.onFocus,
				onBlur: focusProps.onBlur,
				disabled: !isClickable,
				className: cn(stepperIndicatorVariants({
					size,
					variant
				}), !isClickable && "cursor-not-allowed opacity-60", isClickable && "cursor-pointer hover:scale-105", "min-w-[44px] min-h-[44px]"),
				style: indicatorStyles,
				children: renderIndicatorContent()
			}), label && /* @__PURE__ */ jsxs("div", {
				className: "flex flex-col justify-center",
				children: [/* @__PURE__ */ jsx("span", {
					className: cn(stepperLabelVariants({
						size,
						orientation
					})),
					style: labelStyles,
					children: label
				}), description && /* @__PURE__ */ jsx("span", {
					className: cn(stepperDescriptionVariants({
						size,
						orientation
					})),
					style: descriptionStyles,
					children: description
				})]
			})]
		}), !isLast && /* @__PURE__ */ jsx("div", {
			"data-slot": "stepper-separator",
			className: cn(stepperConnectorVariants({ orientation })),
			style: connectorStyles,
			"aria-hidden": "true"
		})]
	});
});
StepperStep.displayName = "StepperStep";
var StepperContent = ({ value, children, className }) => {
	const { value: currentValue, orientation } = useStepperContext();
	const isActive = currentValue === value;
	if (!isActive) return null;
	return /* @__PURE__ */ jsx("div", {
		role: "tabpanel",
		"data-slot": "stepper-content",
		"aria-hidden": !isActive,
		className: cn(stepperContentVariants({ orientation }), className),
		children
	});
};
const StepperGlass = {
	Root: StepperRoot,
	List: StepperList,
	Step: StepperStep,
	Content: StepperContent
};
const SwitchGlass = React.forwardRef(({ className, label, disabled, checked, defaultChecked, ...props }, ref) => {
	const [isChecked, setIsChecked] = React.useState(defaultChecked ?? false);
	const effectiveChecked = checked ?? isChecked;
	const handleCheckedChange = (newChecked) => {
		if (checked === void 0) setIsChecked(newChecked);
		props.onCheckedChange?.(newChecked);
	};
	const switchElement = /* @__PURE__ */ jsx(SwitchPrimitive.Root, {
		ref,
		"data-slot": "switch",
		disabled,
		checked,
		defaultChecked,
		onCheckedChange: handleCheckedChange,
		className: cn("peer inline-flex h-6 w-11 shrink-0 cursor-pointer items-center", "rounded-full border-2 border-transparent", "transition-all duration-300", "focus-visible:outline-none", "disabled:cursor-not-allowed disabled:opacity-50", className),
		style: {
			background: effectiveChecked ? "var(--toggle-active-bg)" : "var(--toggle-bg)",
			boxShadow: effectiveChecked ? "var(--toggle-glow)" : "none"
		},
		...props,
		children: /* @__PURE__ */ jsx(SwitchPrimitive.Thumb, {
			"data-slot": "switch-thumb",
			className: cn("pointer-events-none block h-5 w-5 rounded-full shadow-lg", "ring-0 transition-transform duration-300", "data-[state=checked]:translate-x-5", "data-[state=unchecked]:translate-x-0"),
			style: { background: "var(--toggle-knob)" }
		})
	});
	if (label) return /* @__PURE__ */ jsxs("label", {
		className: cn("inline-flex items-center gap-2.5", disabled ? "opacity-50 cursor-not-allowed" : "cursor-pointer"),
		children: [/* @__PURE__ */ jsx("span", {
			className: "inline-flex items-center justify-center min-h-11",
			children: switchElement
		}), /* @__PURE__ */ jsx("span", {
			className: "text-sm",
			style: { color: "var(--text-secondary)" },
			children: label
		})]
	});
	return switchElement;
});
SwitchGlass.displayName = "SwitchGlass";
const Switch = SwitchGlass;
var SelectGlass = SelectPrimitive.Root;
var SelectGlassGroup = SelectPrimitive.Group;
var SelectGlassValue = SelectPrimitive.Value;
var SelectGlassTrigger = React.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs(SelectPrimitive.Trigger, {
	ref,
	"data-slot": "select-trigger",
	className: cn("flex h-10 w-full items-center justify-between gap-2", "rounded-lg border px-3 py-2 text-sm", "transition-all duration-300", "focus:outline-none", "disabled:cursor-not-allowed disabled:opacity-50", "[&>span]:line-clamp-1", className),
	style: {
		background: "var(--input-bg)",
		borderColor: "var(--input-border)",
		color: "var(--input-text)"
	},
	...props,
	children: [children, /* @__PURE__ */ jsx(SelectPrimitive.Icon, {
		asChild: true,
		children: /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4 opacity-50 shrink-0" })
	})]
}));
SelectGlassTrigger.displayName = "SelectGlassTrigger";
var SelectGlassScrollUpButton = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(SelectPrimitive.ScrollUpButton, {
	ref,
	"data-slot": "select-scroll-up-button",
	className: cn("flex cursor-default items-center justify-center py-1", className),
	style: { color: "var(--text-secondary)" },
	...props,
	children: /* @__PURE__ */ jsx(ChevronUp, { className: "h-4 w-4" })
}));
SelectGlassScrollUpButton.displayName = "SelectGlassScrollUpButton";
var SelectGlassScrollDownButton = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(SelectPrimitive.ScrollDownButton, {
	ref,
	"data-slot": "select-scroll-down-button",
	className: cn("flex cursor-default items-center justify-center py-1", className),
	style: { color: "var(--text-secondary)" },
	...props,
	children: /* @__PURE__ */ jsx(ChevronDown, { className: "h-4 w-4" })
}));
SelectGlassScrollDownButton.displayName = "SelectGlassScrollDownButton";
var SelectGlassContent = React.forwardRef(({ className, children, position = "popper", ...props }, ref) => /* @__PURE__ */ jsx(SelectPrimitive.Portal, { children: /* @__PURE__ */ jsxs(SelectPrimitive.Content, {
	ref,
	"data-slot": "select-content",
	position,
	className: cn(dropdownContentClasses, "p-1", position === "popper" && "data-[side=bottom]:translate-y-1 data-[side=top]:-translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1", className),
	style: getDropdownContentStyles(),
	...props,
	children: [
		/* @__PURE__ */ jsx(SelectGlassScrollUpButton, {}),
		/* @__PURE__ */ jsx(SelectPrimitive.Viewport, {
			className: cn("p-1", position === "popper" && "h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]"),
			children
		}),
		/* @__PURE__ */ jsx(SelectGlassScrollDownButton, {})
	]
}) }));
SelectGlassContent.displayName = "SelectGlassContent";
var SelectGlassLabel = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(SelectPrimitive.Label, {
	ref,
	"data-slot": "select-label",
	className: cn(dropdownLabelClasses, className),
	...props
}));
SelectGlassLabel.displayName = "SelectGlassLabel";
var SelectGlassItem = React.forwardRef(({ className, children, ...props }, ref) => /* @__PURE__ */ jsxs(SelectPrimitive.Item, {
	ref,
	"data-slot": "select-item",
	className: cn(getDropdownItemClasses(), "relative pl-8 pr-2", className),
	...props,
	children: [/* @__PURE__ */ jsx("span", {
		className: "absolute left-2 flex h-3.5 w-3.5 items-center justify-center",
		children: /* @__PURE__ */ jsx(SelectPrimitive.ItemIndicator, { children: /* @__PURE__ */ jsx(Check, { className: "h-4 w-4" }) })
	}), /* @__PURE__ */ jsx(SelectPrimitive.ItemText, { children })]
}));
SelectGlassItem.displayName = "SelectGlassItem";
var SelectGlassSeparator = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ jsx(SelectPrimitive.Separator, {
	ref,
	"data-slot": "select-separator",
	className: cn(dropdownSeparatorClasses, "-mx-1 my-1", className),
	...props
}));
SelectGlassSeparator.displayName = "SelectGlassSeparator";
const Select = SelectGlass;
const SelectGroup = SelectGlassGroup;
const SelectValue = SelectGlassValue;
const SelectTrigger = SelectGlassTrigger;
const SelectScrollUpButton = SelectGlassScrollUpButton;
const SelectScrollDownButton = SelectGlassScrollDownButton;
const SelectContent = SelectGlassContent;
const SelectLabel = SelectGlassLabel;
const SelectItem = SelectGlassItem;
const SelectSeparator = SelectGlassSeparator;
var MOBILE_BREAKPOINT = 768;
var SIDEBAR_COOKIE_NAME = "sidebar:state";
var SIDEBAR_COOKIE_MAX_AGE = 3600 * 24 * 7;
var SidebarContext = createContext(null);
function useSidebar() {
	const context = useContext(SidebarContext);
	if (!context) throw new Error("useSidebar must be used within SidebarGlass.Provider. Wrap your component tree with <SidebarGlass.Provider>.");
	return context;
}
const SidebarProvider = ({ children, open: controlledOpen, onOpenChange, defaultOpen = true, side = "left", variant = "sidebar", collapsible = "offcanvas", cookieName = SIDEBAR_COOKIE_NAME, keyboardShortcut = "b" }) => {
	const [internalOpen, setInternalOpen] = useState(() => {
		if (typeof document !== "undefined") {
			const sidebarCookie = document.cookie.split(";").find((c) => c.trim().startsWith(`${cookieName}=`));
			if (sidebarCookie) return sidebarCookie.split("=")[1] === "true";
		}
		return defaultOpen;
	});
	const isControlled = controlledOpen !== void 0;
	const open = isControlled ? controlledOpen : internalOpen;
	const setOpen = useCallback((value$1) => {
		if (!isControlled) setInternalOpen(value$1);
		onOpenChange?.(value$1);
		if (typeof document !== "undefined") document.cookie = `${cookieName}=${value$1}; path=/; max-age=${SIDEBAR_COOKIE_MAX_AGE}`;
	}, [
		isControlled,
		onOpenChange,
		cookieName
	]);
	const [openMobile, setOpenMobile] = useState(false);
	const [isMobile, setIsMobile] = useState(() => {
		if (typeof window === "undefined") return false;
		return window.innerWidth < MOBILE_BREAKPOINT;
	});
	useEffect(() => {
		if (typeof window === "undefined") return;
		const checkMobile = () => {
			const mobile = window.innerWidth < MOBILE_BREAKPOINT;
			setIsMobile(mobile);
			if (!mobile) setOpenMobile(false);
		};
		checkMobile();
		window.addEventListener("resize", checkMobile);
		return () => window.removeEventListener("resize", checkMobile);
	}, []);
	const toggleSidebar = useCallback(() => {
		if (isMobile) setOpenMobile((prev) => !prev);
		else setOpen(!open);
	}, [
		isMobile,
		open,
		setOpen
	]);
	useEffect(() => {
		if (!keyboardShortcut) return;
		const handleKeyDown = (e) => {
			if (e.key === keyboardShortcut && (e.metaKey || e.ctrlKey)) {
				e.preventDefault();
				toggleSidebar();
			}
		};
		document.addEventListener("keydown", handleKeyDown);
		return () => document.removeEventListener("keydown", handleKeyDown);
	}, [keyboardShortcut, toggleSidebar]);
	const value = useMemo(() => ({
		state: open ? "expanded" : "collapsed",
		open,
		setOpen,
		openMobile,
		setOpenMobile,
		isMobile,
		toggleSidebar,
		side,
		variant,
		collapsible
	}), [
		open,
		setOpen,
		openMobile,
		isMobile,
		toggleSidebar,
		side,
		variant,
		collapsible
	]);
	return /* @__PURE__ */ jsx(SidebarContext.Provider, {
		value,
		children
	});
};
SidebarProvider.displayName = "SidebarGlass.Provider";
const SidebarRoot = forwardRef(({ children, side: sideProp, variant: variantProp, collapsible: collapsibleProp, className, ...props }, ref) => {
	const context = useSidebar();
	const side = sideProp ?? context.side;
	const variant = variantProp ?? context.variant;
	const collapsible = collapsibleProp ?? context.collapsible;
	const { state, open, openMobile, isMobile, setOpenMobile } = context;
	if (isMobile) return /* @__PURE__ */ jsxs(ModalGlass.Root, {
		open: openMobile,
		onOpenChange: setOpenMobile,
		children: [/* @__PURE__ */ jsx(ModalGlass.Overlay, {}), /* @__PURE__ */ jsx("aside", {
			ref,
			"data-sidebar": "sidebar",
			"data-side": side,
			"data-variant": variant,
			"data-collapsible": collapsible,
			"data-state": "expanded",
			"data-mobile": "true",
			className: cn("fixed inset-y-0 z-50 flex flex-col", "w-[var(--sidebar-width-mobile)]", side === "left" ? "left-0" : "right-0", className),
			style: {
				background: "var(--sidebar-bg)",
				color: "var(--sidebar-foreground)",
				borderRight: side === "left" ? "1px solid var(--sidebar-border)" : void 0,
				borderLeft: side === "right" ? "1px solid var(--sidebar-border)" : void 0,
				backdropFilter: "blur(var(--sidebar-backdrop-blur))",
				WebkitBackdropFilter: "blur(var(--sidebar-backdrop-blur))",
				boxShadow: "var(--sidebar-glow)"
			},
			...props,
			children
		})]
	});
	const width = !open && collapsible !== "none" && collapsible === "icon" ? "var(--sidebar-width-icon)" : "var(--sidebar-width)";
	return /* @__PURE__ */ jsx("aside", {
		ref,
		"data-sidebar": "sidebar",
		"data-side": side,
		"data-variant": variant,
		"data-collapsible": collapsible,
		"data-state": state,
		className: cn("group/sidebar relative flex flex-col", "transition-[width] duration-300 ease-in-out", collapsible === "offcanvas" && !open && "w-0 overflow-hidden", variant === "floating" && "rounded-xl m-2", variant === "inset" && "rounded-xl", className),
		style: {
			width: collapsible === "offcanvas" && !open ? 0 : width,
			background: "var(--sidebar-bg)",
			color: "var(--sidebar-foreground)",
			borderRight: side === "left" && variant !== "floating" ? "1px solid var(--sidebar-border)" : void 0,
			borderLeft: side === "right" && variant !== "floating" ? "1px solid var(--sidebar-border)" : void 0,
			border: variant === "floating" ? "1px solid var(--sidebar-border)" : void 0,
			backdropFilter: "blur(var(--sidebar-backdrop-blur))",
			WebkitBackdropFilter: "blur(var(--sidebar-backdrop-blur))",
			boxShadow: variant === "floating" ? "var(--sidebar-glow)" : void 0
		},
		...props,
		children
	});
});
SidebarRoot.displayName = "SidebarGlass.Root";
const SidebarHeader = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "header",
		className: cn("flex shrink-0 flex-col gap-2 p-4", "border-b", className),
		style: { borderColor: "var(--sidebar-border)" },
		...props,
		children
	});
});
SidebarHeader.displayName = "SidebarGlass.Header";
const SidebarContent = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "content",
		className: cn("flex flex-1 flex-col gap-4 overflow-auto p-4", "[&::-webkit-scrollbar]:w-1.5", "[&::-webkit-scrollbar-track]:bg-transparent", "[&::-webkit-scrollbar-thumb]:rounded-full", "[&::-webkit-scrollbar-thumb]:bg-white/10", className),
		...props,
		children
	});
});
SidebarContent.displayName = "SidebarGlass.Content";
const SidebarFooter = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "footer",
		className: cn("flex shrink-0 flex-col gap-2 p-4", "border-t", className),
		style: { borderColor: "var(--sidebar-border)" },
		...props,
		children
	});
});
SidebarFooter.displayName = "SidebarGlass.Footer";
const SidebarRail = forwardRef(({ className, ...props }, ref) => {
	const { toggleSidebar, side } = useSidebar();
	return /* @__PURE__ */ jsx("button", {
		ref,
		"data-sidebar": "rail",
		"aria-label": "Toggle Sidebar",
		tabIndex: -1,
		onClick: toggleSidebar,
		className: cn("absolute inset-y-0 z-20 w-4", "hidden group-hover/sidebar:block", "-translate-x-1/2 transition-all ease-linear", "after:absolute after:inset-y-0 after:left-1/2 after:w-[2px]", "hover:after:bg-[var(--sidebar-border)]", "cursor-col-resize", side === "left" ? "-right-2" : "-left-2", className),
		...props
	});
});
SidebarRail.displayName = "SidebarGlass.Rail";
const SidebarInset = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("main", {
		ref,
		"data-sidebar": "inset",
		className: cn("flex flex-1 flex-col", "min-h-screen", className),
		...props,
		children
	});
});
SidebarInset.displayName = "SidebarGlass.Inset";
const SidebarTrigger = forwardRef(({ asChild = false, className, children, ...props }, ref) => {
	const { toggleSidebar } = useSidebar();
	return /* @__PURE__ */ jsxs(asChild ? Slot : "button", {
		ref,
		"data-sidebar": "trigger",
		onClick: toggleSidebar,
		className: cn("inline-flex items-center justify-center", "h-9 w-9 rounded-lg", "text-sm font-medium", "transition-colors", "text-[var(--sidebar-foreground)]/60", "hover:bg-[var(--sidebar-accent)]", "hover:text-[var(--sidebar-accent-foreground)]", "focus-visible:outline-none focus-visible:ring-2", "focus-visible:ring-[var(--sidebar-ring)]", "disabled:pointer-events-none disabled:opacity-50", className),
		...props,
		children: [children ?? /* @__PURE__ */ jsx(PanelLeft, { className: "h-6 w-6" }), !children && /* @__PURE__ */ jsx("span", {
			className: "sr-only",
			children: "Toggle Sidebar"
		})]
	});
});
SidebarTrigger.displayName = "SidebarGlass.Trigger";
const SidebarSeparator = forwardRef(({ className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("hr", {
		ref,
		"data-sidebar": "separator",
		className: cn("mx-4 my-2 h-px border-0", className),
		style: { background: "var(--sidebar-border)" },
		...props
	});
});
SidebarSeparator.displayName = "SidebarGlass.Separator";
const SidebarGroup = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "group",
		className: cn("flex flex-col gap-2 p-2", className),
		...props,
		children
	});
});
SidebarGroup.displayName = "SidebarGlass.Group";
const SidebarGroupLabel = forwardRef(({ children, asChild = false, className, ...props }, ref) => {
	const { state } = useSidebar();
	return /* @__PURE__ */ jsx(asChild ? Slot : "div", {
		ref,
		"data-sidebar": "group-label",
		"data-state": state,
		className: cn("flex h-8 shrink-0 items-center px-2", "text-xs font-medium text-[var(--sidebar-foreground)]/60", "transition-[margin,opacity] duration-200 ease-linear", "group-data-[state=collapsed]/sidebar:h-0 group-data-[state=collapsed]/sidebar:overflow-hidden", "group-data-[state=collapsed]/sidebar:opacity-0", className),
		...props,
		children
	});
});
SidebarGroupLabel.displayName = "SidebarGlass.GroupLabel";
const SidebarGroupAction = forwardRef(({ asChild = false, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx(asChild ? Slot : "button", {
		ref,
		"data-sidebar": "group-action",
		className: cn("absolute right-2 top-2.5", "flex aspect-square w-5 items-center justify-center", "rounded-md p-0 text-[var(--sidebar-foreground)]/60", "ring-[var(--sidebar-ring)]", "transition-transform hover:bg-[var(--sidebar-accent)]", "hover:text-[var(--sidebar-foreground)]", "focus-visible:outline-none focus-visible:ring-2", "[&>svg]:size-4 [&>svg]:shrink-0", "group-data-[state=collapsed]/sidebar:hidden", "after:absolute after:-inset-2 after:md:hidden", className),
		...props
	});
});
SidebarGroupAction.displayName = "SidebarGlass.GroupAction";
const SidebarGroupContent = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "group-content",
		className: cn("flex flex-col gap-1", className),
		...props,
		children
	});
});
SidebarGroupContent.displayName = "SidebarGlass.GroupContent";
const SidebarMenu = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("ul", {
		ref,
		"data-sidebar": "menu",
		className: cn("flex w-full flex-col gap-1", className),
		...props,
		children
	});
});
SidebarMenu.displayName = "SidebarGlass.Menu";
const SidebarMenuItem = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("li", {
		ref,
		"data-sidebar": "menu-item",
		className: cn("group/menu-item relative text-[var(--sidebar-foreground)]/60", className),
		...props,
		children
	});
});
SidebarMenuItem.displayName = "SidebarGlass.MenuItem";
var menuButtonSizeClasses = {
	default: "h-8 text-sm",
	sm: "h-7 text-xs",
	lg: "h-10 text-sm group-data-[state=collapsed]/sidebar:!p-0"
};
const SidebarMenuButton = forwardRef(({ asChild = false, isActive = false, tooltip, size = "default", variant = "default", className, children, ...props }, ref) => {
	const { state, isMobile } = useSidebar();
	const button = /* @__PURE__ */ jsx(asChild ? Slot : "button", {
		ref,
		"data-sidebar": "menu-button",
		"data-active": isActive,
		"data-size": size,
		className: cn("peer/menu-button flex w-full items-center gap-2", "overflow-hidden rounded-md px-2", "ring-[var(--sidebar-ring)]", "transition-[width,height,padding] duration-200 ease-linear", "focus-visible:outline-none focus-visible:ring-2", "active:bg-[var(--sidebar-accent)] active:text-[var(--sidebar-accent-foreground)]", "disabled:pointer-events-none disabled:opacity-50", "aria-disabled:pointer-events-none aria-disabled:opacity-50", "[&>span:last-child]:truncate", "[&>svg]:size-4 [&>svg]:shrink-0", menuButtonSizeClasses[size], variant === "default" && ["hover:bg-[var(--sidebar-accent)] hover:text-[var(--sidebar-accent-foreground)]"], variant === "outline" && [
			"bg-transparent shadow-none",
			"hover:bg-[var(--sidebar-accent)] hover:text-[var(--sidebar-accent-foreground)]",
			"hover:shadow-[0_0_0_1px_var(--sidebar-border)]"
		], isActive && ["bg-[var(--sidebar-primary)] text-[var(--sidebar-primary-foreground)]", "hover:bg-[var(--sidebar-primary)] hover:text-[var(--sidebar-primary-foreground)]"], "group-data-[state=collapsed]/sidebar:w-8 group-data-[state=collapsed]/sidebar:!px-0", "group-data-[state=collapsed]/sidebar:justify-center", className),
		...props,
		children
	});
	if (!tooltip || isMobile || state === "expanded") return button;
	return /* @__PURE__ */ jsx(TooltipGlassSimple, {
		content: typeof tooltip === "string" ? tooltip : String(tooltip),
		side: "right",
		children: button
	});
});
SidebarMenuButton.displayName = "SidebarGlass.MenuButton";
const SidebarMenuAction = forwardRef(({ asChild = false, showOnHover = false, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx(asChild ? Slot : "button", {
		ref,
		"data-sidebar": "menu-action",
		className: cn("absolute right-1 top-1.5", "flex aspect-square w-5 items-center justify-center", "rounded-md p-0 text-[var(--sidebar-foreground)]/60", "ring-[var(--sidebar-ring)]", "transition-transform hover:bg-[var(--sidebar-accent)]", "hover:text-[var(--sidebar-foreground)]", "focus-visible:outline-none focus-visible:ring-2", "[&>svg]:size-4 [&>svg]:shrink-0", "group-data-[state=collapsed]/sidebar:hidden", showOnHover && "peer-hover/menu-button:opacity-100 group-focus-within/menu-item:opacity-100", showOnHover && "data-[state=open]:opacity-100 md:opacity-0", "after:absolute after:-inset-2 after:md:hidden", className),
		...props
	});
});
SidebarMenuAction.displayName = "SidebarGlass.MenuAction";
const SidebarMenuBadge = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-sidebar": "menu-badge",
		className: cn("pointer-events-none absolute right-1 flex h-5 min-w-5", "select-none items-center justify-center", "rounded-md px-1 text-xs font-medium tabular-nums", "text-[var(--sidebar-foreground)]/60", "group-data-[state=collapsed]/sidebar:hidden", className),
		...props,
		children
	});
});
SidebarMenuBadge.displayName = "SidebarGlass.MenuBadge";
const SidebarMenuSkeleton = forwardRef(({ showIcon = false, className, ...props }, ref) => {
	const width = "70%";
	return /* @__PURE__ */ jsxs("div", {
		ref,
		"data-sidebar": "menu-skeleton",
		className: cn("flex h-8 items-center gap-2 rounded-md px-2", className),
		...props,
		children: [showIcon && /* @__PURE__ */ jsx(SkeletonGlass, { className: "size-4 rounded-md" }), /* @__PURE__ */ jsx(SkeletonGlass, {
			className: "h-4 max-w-[var(--skeleton-width)] flex-1",
			style: { "--skeleton-width": width }
		})]
	});
});
SidebarMenuSkeleton.displayName = "SidebarGlass.MenuSkeleton";
var SidebarMenuSubContext = createContext(null);
const SidebarMenuSub = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx(SidebarMenuSubContext.Provider, {
		value: { open: true },
		children: /* @__PURE__ */ jsx("ul", {
			ref,
			"data-sidebar": "menu-sub",
			className: cn("flex min-w-0 flex-col gap-1", "mx-3.5 border-l border-[var(--sidebar-border)] px-2.5 py-0.5", "group-data-[state=collapsed]/sidebar:hidden", className),
			...props,
			children
		})
	});
});
SidebarMenuSub.displayName = "SidebarGlass.MenuSub";
const SidebarMenuSubItem = forwardRef(({ children, className, ...props }, ref) => {
	return /* @__PURE__ */ jsx("li", {
		ref,
		"data-sidebar": "menu-sub-item",
		className: cn(className),
		...props,
		children
	});
});
SidebarMenuSubItem.displayName = "SidebarGlass.MenuSubItem";
const SidebarMenuSubButton = forwardRef(({ asChild = false, isActive = false, size = "md", className, ...props }, ref) => {
	return /* @__PURE__ */ jsx(asChild ? Slot : "a", {
		ref,
		"data-sidebar": "menu-sub-button",
		"data-active": isActive,
		"data-size": size,
		className: cn("flex min-w-0 items-center gap-2", "-ml-px rounded-md border-l border-transparent", "text-[var(--sidebar-foreground)]/60", "ring-[var(--sidebar-ring)]", "transition-colors", "hover:border-[var(--sidebar-border)]", "hover:bg-[var(--sidebar-accent)]", "hover:text-[var(--sidebar-accent-foreground)]", "focus-visible:outline-none focus-visible:ring-2", "active:bg-[var(--sidebar-accent)] active:text-[var(--sidebar-accent-foreground)]", "disabled:pointer-events-none disabled:opacity-50", "[&>span:last-child]:truncate", "[&>svg]:size-4 [&>svg]:shrink-0", size === "sm" && "h-7 px-2 text-xs", size === "md" && "h-8 px-2 text-sm", isActive && ["border-[var(--sidebar-primary)]", "text-[var(--sidebar-foreground)]"], className),
		...props
	});
});
SidebarMenuSubButton.displayName = "SidebarGlass.MenuSubButton";
const SidebarGlass = {
	Provider: SidebarProvider,
	Root: SidebarRoot,
	Header: SidebarHeader,
	Content: SidebarContent,
	Footer: SidebarFooter,
	Rail: SidebarRail,
	Inset: SidebarInset,
	Trigger: SidebarTrigger,
	Separator: SidebarSeparator,
	Group: SidebarGroup,
	GroupLabel: SidebarGroupLabel,
	GroupAction: SidebarGroupAction,
	GroupContent: SidebarGroupContent,
	Menu: SidebarMenu,
	MenuItem: SidebarMenuItem,
	MenuButton: SidebarMenuButton,
	MenuAction: SidebarMenuAction,
	MenuBadge: SidebarMenuBadge,
	MenuSkeleton: SidebarMenuSkeleton,
	MenuSub: SidebarMenuSub,
	MenuSubItem: SidebarMenuSubItem,
	MenuSubButton: SidebarMenuSubButton
};
const AnimatedBackgroundGlass = forwardRef(({ className, showCenterOrb, ...props }, ref) => {
	const { theme } = useTheme();
	const shouldShowCenterOrb = showCenterOrb ?? theme === "glass";
	return /* @__PURE__ */ jsxs("div", {
		ref,
		className: cn("fixed inset-0 transition-all duration-500 overflow-hidden", className),
		style: { background: "linear-gradient(135deg, var(--bg-from), var(--bg-via), var(--bg-to))" },
		"aria-hidden": "true",
		...props,
		children: [
			/* @__PURE__ */ jsx("div", {
				className: "absolute top-20 -left-20 w-[600px] h-[600px] rounded-full animate-orb-float",
				style: {
					background: "var(--orb-1)",
					filter: "blur(80px)"
				}
			}),
			/* @__PURE__ */ jsx("div", {
				className: "absolute -bottom-20 -right-20 w-[700px] h-[700px] rounded-full animate-orb-float",
				style: {
					background: "var(--orb-2)",
					filter: "blur(100px)",
					animationDelay: "2s"
				}
			}),
			/* @__PURE__ */ jsx("div", {
				className: "absolute top-1/3 -right-10 w-[500px] h-[500px] rounded-full animate-orb-float",
				style: {
					background: "var(--orb-3)",
					filter: "blur(70px)",
					animationDelay: "4s"
				}
			}),
			/* @__PURE__ */ jsx("div", {
				className: "absolute -bottom-10 left-1/4 w-[450px] h-[450px] rounded-full animate-orb-float",
				style: {
					background: "var(--orb-4)",
					filter: "blur(60px)",
					animationDelay: "6s"
				}
			}),
			shouldShowCenterOrb && /* @__PURE__ */ jsx("div", {
				className: "absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 w-[500px] h-[500px] rounded-full",
				style: {
					background: "var(--orb-5)",
					filter: "blur(80px)"
				}
			})
		]
	});
});
AnimatedBackgroundGlass.displayName = "AnimatedBackgroundGlass";
const AnimatedBackground = AnimatedBackgroundGlass;
function ScrollArea({ className, children, ...props }) {
	return /* @__PURE__ */ jsxs(ScrollAreaPrimitive.Root, {
		"data-slot": "scroll-area",
		className: cn("relative", className),
		...props,
		children: [
			/* @__PURE__ */ jsx(ScrollAreaPrimitive.Viewport, {
				"data-slot": "scroll-area-viewport",
				className: "focus-visible:ring-ring/50 size-full rounded-[inherit] transition-[color,box-shadow] outline-none focus-visible:ring-[3px] focus-visible:outline-1",
				children
			}),
			/* @__PURE__ */ jsx(ScrollBar, {}),
			/* @__PURE__ */ jsx(ScrollAreaPrimitive.Corner, {})
		]
	});
}
function ScrollBar({ className, orientation = "vertical", ...props }) {
	return /* @__PURE__ */ jsx(ScrollAreaPrimitive.ScrollAreaScrollbar, {
		"data-slot": "scroll-area-scrollbar",
		orientation,
		className: cn("flex touch-none p-px transition-colors select-none", orientation === "vertical" && "h-full w-2.5 border-l border-l-transparent", orientation === "horizontal" && "h-2.5 flex-col border-t border-t-transparent", className),
		...props,
		children: /* @__PURE__ */ jsx(ScrollAreaPrimitive.ScrollAreaThumb, {
			"data-slot": "scroll-area-thumb",
			className: "bg-border relative flex-1 rounded-full"
		})
	});
}
var SplitLayoutContext = createContext(null);
function useSplitLayout() {
	const context = useContext(SplitLayoutContext);
	if (!context) throw new Error("useSplitLayout must be used within SplitLayoutGlass.Provider. Wrap your component tree with <SplitLayoutGlass.Provider>.");
	return context;
}
function useSplitLayoutOptional() {
	return useContext(SplitLayoutContext);
}
var BREAKPOINT_VALUES = {
	sm: 640,
	md: 768,
	lg: 1024,
	xl: 1440,
	"2xl": 1536
};
const SplitLayoutProvider = ({ children, selectedKey: controlledSelectedKey, onSelectedKeyChange, defaultSelectedKey = null, open: controlledOpen, onOpenChange, defaultOpen = true, breakpoint = "md", mobileMode = "stack", intensity = "medium", stickyOffset = 24, urlParamName, keyboardShortcut = "b" }) => {
	const [internalSelectedKey, setInternalSelectedKey] = useState(() => {
		if (urlParamName && typeof window !== "undefined") {
			const urlValue = new URLSearchParams(window.location.search).get(urlParamName);
			if (urlValue) return urlValue;
		}
		return defaultSelectedKey;
	});
	const isSelectedKeyControlled = controlledSelectedKey !== void 0;
	const selectedKey = isSelectedKeyControlled ? controlledSelectedKey : internalSelectedKey;
	const setSelectedKey = useCallback((key) => {
		if (!isSelectedKeyControlled) setInternalSelectedKey(key);
		onSelectedKeyChange?.(key);
		if (urlParamName && typeof window !== "undefined") {
			const url = new URL(window.location.href);
			if (key) url.searchParams.set(urlParamName, key);
			else url.searchParams.delete(urlParamName);
			window.history.replaceState({}, "", url.toString());
		}
	}, [
		isSelectedKeyControlled,
		onSelectedKeyChange,
		urlParamName
	]);
	const [internalOpen, setInternalOpen] = useState(defaultOpen);
	const isOpenControlled = controlledOpen !== void 0;
	const isOpen = isOpenControlled ? controlledOpen : internalOpen;
	const setIsOpen = useCallback((open) => {
		if (!isOpenControlled) setInternalOpen(open);
		onOpenChange?.(open);
	}, [isOpenControlled, onOpenChange]);
	const [isMobileOpen, setMobileOpen] = useState(false);
	const [isMobile, setIsMobile] = useState(() => {
		if (typeof window === "undefined") return false;
		return window.innerWidth < BREAKPOINT_VALUES[breakpoint];
	});
	useEffect(() => {
		if (typeof window === "undefined") return;
		const checkMobile = () => {
			setIsMobile(window.innerWidth < BREAKPOINT_VALUES[breakpoint]);
		};
		checkMobile();
		window.addEventListener("resize", checkMobile);
		return () => window.removeEventListener("resize", checkMobile);
	}, [breakpoint]);
	const toggle = useCallback(() => {
		if (isMobile) setMobileOpen((prev) => !prev);
		else setIsOpen(!isOpen);
	}, [
		isMobile,
		isOpen,
		setIsOpen
	]);
	useEffect(() => {
		if (!keyboardShortcut) return;
		const handleKeyDown = (e) => {
			if (e.key === "Escape" && isMobile && isMobileOpen) {
				e.preventDefault();
				setMobileOpen(false);
				return;
			}
			if (e.key === keyboardShortcut && (e.metaKey || e.ctrlKey)) {
				e.preventDefault();
				toggle();
			}
		};
		document.addEventListener("keydown", handleKeyDown);
		return () => document.removeEventListener("keydown", handleKeyDown);
	}, [
		keyboardShortcut,
		isMobile,
		isMobileOpen,
		toggle
	]);
	const value = useMemo(() => ({
		selectedKey,
		setSelectedKey,
		isOpen,
		setIsOpen,
		isMobileOpen,
		setMobileOpen,
		isMobile,
		breakpoint,
		mobileMode,
		intensity,
		stickyOffset,
		toggle,
		state: isOpen ? "expanded" : "collapsed",
		open: isOpen,
		setOpen: setIsOpen,
		openMobile: isMobileOpen,
		setOpenMobile: setMobileOpen,
		toggleSidebar: toggle
	}), [
		selectedKey,
		setSelectedKey,
		isOpen,
		setIsOpen,
		isMobileOpen,
		isMobile,
		breakpoint,
		mobileMode,
		intensity,
		stickyOffset,
		toggle
	]);
	return /* @__PURE__ */ jsx(SplitLayoutContext.Provider, {
		value,
		children
	});
};
SplitLayoutProvider.displayName = "SplitLayoutGlass.Provider";
var SplitLayoutAccordionRoot = forwardRef(({ children, label = "Content accordion", className, ...props }, ref) => {
	return /* @__PURE__ */ jsx(GlassCard, {
		asChild: true,
		intensity: useSplitLayoutOptional()?.intensity ?? "medium",
		padding: "none",
		className: cn("divide-y divide-white/10", className),
		children: /* @__PURE__ */ jsx("div", {
			ref,
			role: "region",
			"aria-label": label,
			"data-split-accordion": "",
			...props,
			children
		})
	});
});
SplitLayoutAccordionRoot.displayName = "SplitLayoutAccordion.Root";
var SplitLayoutAccordionItem = forwardRef(({ itemKey, trigger, children, disabled = false, className, ...props }, ref) => {
	const { selectedKey, setSelectedKey } = useSplitLayout();
	const isExpanded = selectedKey === itemKey;
	const contentRef = useRef(null);
	const [contentHeight, setContentHeight] = useState(null);
	const triggerId = useId();
	const contentId = useId();
	useEffect(() => {
		if (contentRef.current) setContentHeight(contentRef.current.scrollHeight);
	}, [children, isExpanded]);
	const handleToggle = useCallback(() => {
		if (disabled) return;
		setSelectedKey(isExpanded ? null : itemKey);
	}, [
		disabled,
		isExpanded,
		itemKey,
		setSelectedKey
	]);
	const handleKeyDown = useCallback((e) => {
		if (disabled) return;
		switch (e.key) {
			case "Enter":
			case " ":
				e.preventDefault();
				handleToggle();
				break;
			case "ArrowDown":
				e.preventDefault();
				((e.currentTarget.parentElement?.nextElementSibling)?.querySelector("[data-accordion-trigger]"))?.focus();
				break;
			case "ArrowUp":
				e.preventDefault();
				((e.currentTarget.parentElement?.previousElementSibling)?.querySelector("[data-accordion-trigger]"))?.focus();
				break;
			case "Home":
				e.preventDefault();
				(e.currentTarget.closest("[data-split-accordion]")?.querySelector("[data-accordion-trigger]"))?.focus();
				break;
			case "End": {
				e.preventDefault();
				const triggers = e.currentTarget.closest("[data-split-accordion]")?.querySelectorAll("[data-accordion-trigger]");
				(triggers?.[triggers.length - 1])?.focus();
				break;
			}
		}
	}, [disabled, handleToggle]);
	return /* @__PURE__ */ jsxs("div", {
		ref,
		"data-accordion-item": "",
		"data-state": isExpanded ? "open" : "closed",
		"data-disabled": disabled || void 0,
		className: cn("overflow-hidden", disabled && "opacity-50 cursor-not-allowed", className),
		...props,
		children: [/* @__PURE__ */ jsxs("button", {
			type: "button",
			id: triggerId,
			"data-accordion-trigger": "",
			"aria-expanded": isExpanded,
			"aria-controls": contentId,
			"aria-disabled": disabled,
			disabled,
			onClick: handleToggle,
			onKeyDown: handleKeyDown,
			className: cn("flex w-full items-center justify-between gap-4", "p-4 text-left", "transition-colors duration-200", "hover:bg-white/5 focus-visible:bg-white/5", "focus:outline-none focus-visible:ring-2 focus-visible:ring-white/20 focus-visible:ring-inset", disabled && "pointer-events-none"),
			children: [/* @__PURE__ */ jsx("div", {
				className: "flex-1",
				children: trigger
			}), /* @__PURE__ */ jsx(ChevronDown, { className: cn("h-4 w-4 shrink-0 text-white/60", "transition-transform duration-300 ease-out", isExpanded && "rotate-180") })]
		}), /* @__PURE__ */ jsx("div", {
			id: contentId,
			ref: contentRef,
			role: "region",
			"aria-labelledby": triggerId,
			"data-accordion-content": "",
			"data-state": isExpanded ? "open" : "closed",
			className: cn("overflow-hidden", "transition-[max-height,opacity] duration-300 ease-out"),
			style: {
				maxHeight: isExpanded ? contentHeight ?? "auto" : 0,
				opacity: isExpanded ? 1 : 0
			},
			children: /* @__PURE__ */ jsx("div", {
				className: "p-4 pt-0 border-t border-white/5",
				children
			})
		})]
	});
});
SplitLayoutAccordionItem.displayName = "SplitLayoutAccordion.Item";
const SplitLayoutAccordion = {
	Root: SplitLayoutAccordionRoot,
	Item: SplitLayoutAccordionItem
};
function useGridStyles(config) {
	const { ratio = {
		sidebar: 1,
		main: 2
	}, minSidebarWidth = "280px", maxSidebarWidth, gap = {
		mobile: 16,
		desktop: 24
	}, stickyOffset = 24 } = config;
	const gapMobile = typeof gap === "number" ? gap : gap.mobile ?? 16;
	const gapDesktop = typeof gap === "number" ? gap : gap.desktop ?? 24;
	const gridTemplate = maxSidebarWidth ? `minmax(${minSidebarWidth}, ${maxSidebarWidth}) ${ratio.main}fr` : `minmax(${minSidebarWidth}, ${ratio.sidebar}fr) ${ratio.main}fr`;
	return { cssVars: {
		"--grid-template": gridTemplate,
		"--sticky-offset": `${stickyOffset}px`,
		"--sticky-max-height": `calc(100vh - calc(${stickyOffset}px * 2))`,
		"--gap-mobile": `${gapMobile}px`,
		"--gap-desktop": `${gapDesktop}px`
	} };
}
var SplitLayoutRoot = forwardRef(({ children, ratio = {
	sidebar: 1,
	main: 2
}, minSidebarWidth = "300px", maxSidebarWidth, gap = {
	mobile: 16,
	desktop: 24
}, breakpoint: breakpointProp, mobileLayout = "stack", className, ...props }, ref) => {
	const context = useSplitLayoutOptional();
	const breakpoint = breakpointProp ?? context?.breakpoint ?? "md";
	const { cssVars } = useGridStyles({
		ratio,
		minSidebarWidth,
		maxSidebarWidth,
		gap,
		stickyOffset: context?.stickyOffset ?? 24
	});
	const bp = breakpoint;
	const gapClasses = {
		sm: "gap-[var(--gap-mobile)] sm:gap-[var(--gap-desktop)]",
		md: "gap-[var(--gap-mobile)] md:gap-[var(--gap-desktop)]",
		lg: "gap-[var(--gap-mobile)] lg:gap-[var(--gap-desktop)]",
		xl: "gap-[var(--gap-mobile)] xl:gap-[var(--gap-desktop)]",
		"2xl": "gap-[var(--gap-mobile)] 2xl:gap-[var(--gap-desktop)]"
	};
	const gridClasses = {
		sm: "sm:grid-cols-(--grid-template)",
		md: "md:grid-cols-(--grid-template)",
		lg: "lg:grid-cols-(--grid-template)",
		xl: "xl:grid-cols-(--grid-template)",
		"2xl": "2xl:grid-cols-(--grid-template)"
	};
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-split-layout-root": "",
		"data-state": context?.state ?? "expanded",
		className: cn("grid", "items-start", mobileLayout === "stack" && "grid-cols-1", mobileLayout === "main-only" && "grid-cols-1 *:data-split-sidebar:hidden", mobileLayout === "sidebar-only" && "grid-cols-1 *:data-split-main:hidden", gridClasses[bp], gapClasses[bp], className),
		style: cssVars,
		...props,
		children
	});
});
SplitLayoutRoot.displayName = "SplitLayoutGlass.Root";
var SplitLayoutSidebar = forwardRef(({ children, label = "Sidebar navigation", className, ...props }, ref) => {
	const context = useSplitLayoutOptional();
	const breakpoint = context?.breakpoint ?? "md";
	const intensity = context?.intensity ?? "medium";
	const bp = breakpoint;
	return /* @__PURE__ */ jsx(GlassCard, {
		asChild: true,
		intensity,
		padding: "none",
		className: cn("overflow-hidden rounded-xl", `${bp}:sticky`, `${bp}:top-[var(--sticky-offset)]`, `${bp}:max-h-[var(--sticky-max-height)]`, `${bp}:flex`, `${bp}:flex-col`, className),
		children: /* @__PURE__ */ jsx("aside", {
			ref,
			"data-split-sidebar": "",
			"aria-label": label,
			...props,
			children
		})
	});
});
SplitLayoutSidebar.displayName = "SplitLayoutGlass.Sidebar";
var SplitLayoutSidebarHeader = forwardRef(({ children, className, ...props }, ref) => /* @__PURE__ */ jsx("div", {
	ref,
	"data-split-sidebar-header": "",
	className: cn("shrink-0 p-4 border-b border-white/10", className),
	...props,
	children
}));
SplitLayoutSidebarHeader.displayName = "SplitLayoutGlass.SidebarHeader";
var SplitLayoutSidebarContent = forwardRef(({ children, scrollable = true, className, ...props }, ref) => {
	if (scrollable) {
		const { dir, ...divProps } = props;
		return /* @__PURE__ */ jsx(ScrollArea, {
			ref,
			"data-split-sidebar-content": "",
			className: cn("flex-1 min-h-0", className),
			...divProps,
			children: /* @__PURE__ */ jsx("div", {
				className: "p-4",
				children
			})
		});
	}
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-split-sidebar-content": "",
		className: cn("flex-1 min-h-0 overflow-auto p-4", className),
		...props,
		children
	});
});
SplitLayoutSidebarContent.displayName = "SplitLayoutGlass.SidebarContent";
var SplitLayoutSidebarFooter = forwardRef(({ children, className, ...props }, ref) => /* @__PURE__ */ jsx("div", {
	ref,
	"data-split-sidebar-footer": "",
	className: cn("shrink-0 p-4 border-t border-white/10", className),
	...props,
	children
}));
SplitLayoutSidebarFooter.displayName = "SplitLayoutGlass.SidebarFooter";
var SplitLayoutMain = forwardRef(({ children, label = "Main content", className, ...props }, ref) => {
	const context = useSplitLayoutOptional();
	const breakpoint = context?.breakpoint ?? "md";
	const intensity = context?.intensity ?? "medium";
	const bp = breakpoint;
	return /* @__PURE__ */ jsx(GlassCard, {
		asChild: true,
		intensity,
		padding: "none",
		className: cn("overflow-hidden rounded-xl", `${bp}:sticky`, `${bp}:top-[var(--sticky-offset)]`, `${bp}:max-h-[var(--sticky-max-height)]`, `${bp}:flex`, `${bp}:flex-col`, className),
		children: /* @__PURE__ */ jsx("main", {
			ref,
			"data-split-main": "",
			"aria-label": label,
			...props,
			children
		})
	});
});
SplitLayoutMain.displayName = "SplitLayoutGlass.Main";
var SplitLayoutMainHeader = forwardRef(({ children, className, ...props }, ref) => /* @__PURE__ */ jsx("div", {
	ref,
	"data-split-main-header": "",
	className: cn("shrink-0 p-6 border-b border-white/10", className),
	...props,
	children
}));
SplitLayoutMainHeader.displayName = "SplitLayoutGlass.MainHeader";
var SplitLayoutMainContent = forwardRef(({ children, scrollable = true, className, ...props }, ref) => {
	if (scrollable) {
		const { dir, ...divProps } = props;
		return /* @__PURE__ */ jsx(ScrollArea, {
			ref,
			"data-split-main-content": "",
			className: cn("flex-1 min-h-0", className),
			...divProps,
			children: /* @__PURE__ */ jsx("div", {
				className: "p-6",
				children
			})
		});
	}
	return /* @__PURE__ */ jsx("div", {
		ref,
		"data-split-main-content": "",
		className: cn("flex-1 min-h-0 overflow-auto p-6", className),
		...props,
		children
	});
});
SplitLayoutMainContent.displayName = "SplitLayoutGlass.MainContent";
var SplitLayoutMainFooter = forwardRef(({ children, className, ...props }, ref) => /* @__PURE__ */ jsx("div", {
	ref,
	"data-split-main-footer": "",
	className: cn("shrink-0 p-6 border-t border-white/10", className),
	...props,
	children
}));
SplitLayoutMainFooter.displayName = "SplitLayoutGlass.MainFooter";
var SplitLayoutTrigger = forwardRef(({ asChild = false, showOnDesktop = false, variant = "menu", className, children, ...props }, ref) => {
	const { toggle, isOpen, isMobileOpen, isMobile, breakpoint } = useSplitLayout();
	const currentOpen = isMobile ? isMobileOpen : isOpen;
	const Icon = variant === "menu" ? currentOpen ? X : Menu : currentOpen ? PanelLeftClose : PanelLeftOpen;
	const visibilityClass = showOnDesktop ? "" : `${breakpoint}:hidden`;
	if (asChild) return /* @__PURE__ */ jsx(Slot, {
		ref,
		onClick: toggle,
		"aria-label": currentOpen ? "Close sidebar" : "Open sidebar",
		"aria-expanded": currentOpen,
		"data-state": currentOpen ? "open" : "closed",
		className: cn(visibilityClass, className),
		...props,
		children
	});
	return /* @__PURE__ */ jsx(ButtonGlass, {
		ref,
		variant: "ghost",
		size: "icon",
		onClick: toggle,
		"aria-label": currentOpen ? "Close sidebar" : "Open sidebar",
		"aria-expanded": currentOpen,
		"data-state": currentOpen ? "open" : "closed",
		className: cn(visibilityClass, className),
		...props,
		children: children ?? /* @__PURE__ */ jsx(Icon, { className: "h-5 w-5" })
	});
});
SplitLayoutTrigger.displayName = "SplitLayoutGlass.Trigger";
const SplitLayoutGlass = {
	Provider: SplitLayoutProvider,
	Root: SplitLayoutRoot,
	Sidebar: SplitLayoutSidebar,
	SidebarHeader: SplitLayoutSidebarHeader,
	SidebarContent: SplitLayoutSidebarContent,
	SidebarFooter: SplitLayoutSidebarFooter,
	Main: SplitLayoutMain,
	MainHeader: SplitLayoutMainHeader,
	MainContent: SplitLayoutMainContent,
	MainFooter: SplitLayoutMainFooter,
	Trigger: SplitLayoutTrigger
};
var ChartContext = React.createContext(null);
function useChart() {
	const context = React.useContext(ChartContext);
	if (!context) throw new Error("useChart must be used within a <ChartContainer />");
	return context;
}
var ChartContainer = React.forwardRef(({ id, className, children, config, ...props }, ref) => {
	const uniqueId = React.useId();
	const chartId = `chart-${id || uniqueId.replace(/:/g, "")}`;
	return /* @__PURE__ */ jsx(ChartContext.Provider, {
		value: { config },
		children: /* @__PURE__ */ jsxs("div", {
			"data-chart": chartId,
			ref,
			className: cn("flex aspect-video justify-center text-xs [&_.recharts-cartesian-axis-tick_text]:fill-[var(--text-muted)] [&_.recharts-cartesian-grid_line[stroke='#ccc']]:stroke-[var(--glass-border)] [&_.recharts-curve.recharts-tooltip-cursor]:stroke-[var(--glass-border)] [&_.recharts-dot[stroke='#fff']]:stroke-transparent [&_.recharts-layer]:outline-none [&_.recharts-polar-grid_[stroke='#ccc']]:stroke-[var(--glass-border)] [&_.recharts-radial-bar-background-sector]:fill-[var(--glass-frost-5)] [&_.recharts-rectangle.recharts-tooltip-cursor]:fill-[var(--glass-frost-10)] [&_.recharts-reference-line_[stroke='#ccc']]:stroke-[var(--glass-border)] [&_.recharts-sector[stroke='#fff']]:stroke-transparent [&_.recharts-sector]:outline-none [&_.recharts-surface]:outline-none", className),
			...props,
			children: [/* @__PURE__ */ jsx(ChartStyle, {
				id: chartId,
				config
			}), /* @__PURE__ */ jsx(RechartsPrimitive.ResponsiveContainer, { children })]
		})
	});
});
ChartContainer.displayName = "ChartContainer";
var ChartStyle = ({ id, config }) => {
	const colorConfig = Object.entries(config).filter(([, itemConfig]) => itemConfig.color || itemConfig.theme);
	if (!colorConfig.length) return null;
	return /* @__PURE__ */ jsx("style", { dangerouslySetInnerHTML: { __html: `
[data-chart="${id}"] {
${colorConfig.map(([key, itemConfig]) => {
		const color = itemConfig.color || itemConfig.theme?.light;
		return color ? `  --color-${key}: ${color};` : null;
	}).filter(Boolean).join("\n")}
}

[data-theme="glass"] [data-chart="${id}"],
.dark [data-chart="${id}"] {
${colorConfig.map(([key, itemConfig]) => {
		const color = itemConfig.theme?.dark || itemConfig.color;
		return color ? `  --color-${key}: ${color};` : null;
	}).filter(Boolean).join("\n")}
}
` } });
};
var ChartTooltip = RechartsPrimitive.Tooltip;
var ChartTooltipContent = React.forwardRef(({ active, payload, className, indicator = "dot", hideLabel = false, hideIndicator = false, label, labelFormatter, labelClassName, nameKey, labelKey, valueFormatter }, ref) => {
	const { config } = useChart();
	const tooltipLabel = React.useMemo(() => {
		if (hideLabel || !payload?.length) return null;
		const [item] = payload;
		const itemConfig = getPayloadConfigFromPayload(config, item, `${labelKey || item?.dataKey || item?.name || "value"}`);
		const value = !labelKey && typeof label === "string" ? config[label]?.label || label : itemConfig?.label;
		if (labelFormatter && typeof label === "string") return /* @__PURE__ */ jsx("div", {
			className: cn("font-medium", labelClassName),
			children: labelFormatter(label, payload)
		});
		if (!value) return null;
		return /* @__PURE__ */ jsx("div", {
			className: cn("font-medium", labelClassName),
			children: value
		});
	}, [
		label,
		labelFormatter,
		payload,
		hideLabel,
		labelClassName,
		config,
		labelKey
	]);
	if (!active || !payload?.length) return null;
	const nestLabel = payload.length === 1 && indicator !== "dot";
	return /* @__PURE__ */ jsxs("div", {
		ref,
		className: cn("grid min-w-[8rem] items-start gap-1.5 rounded-lg border border-[var(--glass-border)] bg-[var(--glass-bg)] px-2.5 py-1.5 text-xs shadow-xl backdrop-blur-md", className),
		children: [!nestLabel ? tooltipLabel : null, /* @__PURE__ */ jsx("div", {
			className: "grid gap-1.5",
			children: payload.map((item, index) => {
				const itemConfig = getPayloadConfigFromPayload(config, item, `${nameKey || item.name || item.dataKey || "value"}`);
				const indicatorColor = item.fill || item.color;
				return /* @__PURE__ */ jsxs("div", {
					className: cn("flex w-full flex-wrap items-stretch gap-2 [&>svg]:h-2.5 [&>svg]:w-2.5 [&>svg]:text-[var(--text-muted)]", indicator === "dot" && "items-center"),
					children: [itemConfig?.icon ? /* @__PURE__ */ jsx(itemConfig.icon, {}) : !hideIndicator && /* @__PURE__ */ jsx("div", {
						className: cn("shrink-0 rounded-[2px] border-[--color-border] bg-[--color-bg]", {
							"h-2.5 w-2.5": indicator === "dot",
							"w-1": indicator === "line",
							"w-0 border-[1.5px] border-dashed bg-transparent": indicator === "dashed",
							"my-0.5": nestLabel && indicator === "dashed"
						}),
						style: {
							"--color-bg": indicatorColor,
							"--color-border": indicatorColor
						}
					}), /* @__PURE__ */ jsxs("div", {
						className: cn("flex flex-1 justify-between leading-none", nestLabel ? "items-end" : "items-center"),
						children: [/* @__PURE__ */ jsxs("div", {
							className: "grid gap-1.5",
							children: [nestLabel ? tooltipLabel : null, /* @__PURE__ */ jsx("span", {
								className: "text-[var(--text-muted)]",
								children: itemConfig?.label || item.name
							})]
						}), item.value !== void 0 && /* @__PURE__ */ jsx("span", {
							className: "font-mono font-medium tabular-nums text-[var(--text-primary)]",
							children: valueFormatter ? valueFormatter(item.value) : item.value.toLocaleString()
						})]
					})]
				}, `${item.dataKey}-${index}`);
			})
		})]
	});
});
ChartTooltipContent.displayName = "ChartTooltipContent";
var ChartLegend = RechartsPrimitive.Legend;
var ChartLegendContent = React.forwardRef(({ className, hideIcon = false, payload, verticalAlign = "bottom", nameKey }, ref) => {
	const { config } = useChart();
	if (!payload?.length) return null;
	return /* @__PURE__ */ jsx("div", {
		ref,
		className: cn("flex items-center justify-center gap-4", verticalAlign === "top" ? "pb-3" : "pt-3", className),
		children: payload.map((item) => {
			const itemConfig = getPayloadConfigFromPayload(config, item, `${nameKey || item.dataKey || "value"}`);
			return /* @__PURE__ */ jsxs("div", {
				className: cn("flex items-center gap-1.5 [&>svg]:h-3 [&>svg]:w-3 [&>svg]:text-[var(--text-muted)]"),
				children: [itemConfig?.icon && !hideIcon ? /* @__PURE__ */ jsx(itemConfig.icon, {}) : /* @__PURE__ */ jsx("div", {
					className: "h-2 w-2 shrink-0 rounded-[2px]",
					style: { backgroundColor: item.color }
				}), itemConfig?.label]
			}, item.value);
		})
	});
});
ChartLegendContent.displayName = "ChartLegendContent";
function getPayloadConfigFromPayload(config, payload, key) {
	if (typeof payload !== "object" || payload === null) return;
	const payloadPayload = "payload" in payload && typeof payload.payload === "object" && payload.payload !== null ? payload.payload : void 0;
	let configLabelKey = key;
	if (key in config) return config[key];
	if (payloadPayload && key in payloadPayload) configLabelKey = payloadPayload[key];
	return configLabelKey in config ? config[configLabelKey] : config[key];
}
const dropdownAlign = cva("absolute mt-2 min-w-[160px] md:min-w-[200px] rounded-2xl py-1.5 md:py-2", {
	variants: { align: {
		left: "left-0",
		right: "right-0"
	} },
	defaultVariants: { align: "left" }
});
const tooltipPositions = cva("absolute z-50 px-2 py-1 md:px-3 md:py-1.5 rounded-lg text-[10px] md:text-xs font-medium whitespace-nowrap transition-opacity duration-200", {
	variants: { position: {
		top: "bottom-full left-1/2 -translate-x-1/2 mb-2",
		bottom: "top-full left-1/2 -translate-x-1/2 mt-2",
		left: "right-full top-1/2 -translate-y-1/2 mr-2",
		right: "left-full top-1/2 -translate-y-1/2 ml-2"
	} },
	defaultVariants: { position: "top" }
});
const alertVariants$1 = cva("relative w-full rounded-lg border px-4 py-3 text-sm grid has-[>svg]:grid-cols-[calc(var(--spacing)*4)_1fr] grid-cols-[0_1fr] has-[>svg]:gap-x-3 gap-y-0.5 items-start [&>svg]:size-4 [&>svg]:translate-y-0.5 [&>svg]:text-current", {
	variants: { variant: {
		default: "bg-card text-card-foreground",
		destructive: "text-destructive bg-card [&>svg]:text-current *:data-[slot=alert-description]:text-destructive/90"
	} },
	defaultVariants: { variant: "default" }
});
const badgeVariants$1 = cva("inline-flex items-center justify-center rounded-full border px-2 py-0.5 text-xs font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden", {
	variants: { variant: {
		default: "border-transparent bg-primary text-primary-foreground [a&]:hover:bg-primary/90",
		secondary: "border-transparent bg-secondary text-secondary-foreground [a&]:hover:bg-secondary/90",
		destructive: "border-transparent bg-destructive text-white [a&]:hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60",
		outline: "text-foreground [a&]:hover:bg-accent [a&]:hover:text-accent-foreground"
	} },
	defaultVariants: { variant: "default" }
});
const buttonVariants = cva("inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive", {
	variants: {
		variant: {
			default: "bg-primary text-primary-foreground hover:bg-primary/90",
			destructive: "bg-destructive text-white hover:bg-destructive/90 focus-visible:ring-destructive/20 dark:focus-visible:ring-destructive/40 dark:bg-destructive/60",
			outline: "border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50",
			secondary: "bg-secondary text-secondary-foreground hover:bg-secondary/80",
			ghost: "hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50",
			link: "text-primary underline-offset-4 hover:underline"
		},
		size: {
			default: "h-9 px-4 py-2 has-[>svg]:px-3",
			sm: "h-8 rounded-md gap-1.5 px-3 has-[>svg]:px-2.5",
			lg: "h-10 rounded-md px-6 has-[>svg]:px-4",
			icon: "size-9",
			"icon-sm": "size-8",
			"icon-lg": "size-10"
		}
	},
	defaultVariants: {
		variant: "default",
		size: "default"
	}
});
export { AICardGlass, Alert, AlertDescription, AlertGlass, AlertGlassDescription, AlertGlassTitle, AlertTitle, AnimatedBackground, AnimatedBackgroundGlass, Avatar, AvatarFallback, AvatarGlass, AvatarGlassFallback, AvatarGlassImage, AvatarGlassSimple, AvatarImage, Badge, BadgeGlass, Bar, BarChart, BaseProgressGlass, Button, ButtonGlass, Card, CardContent, CardDescription, CardFooter, CardGlass, CardGlassAction, CardGlassContent, CardGlassDescription, CardGlassFooter, CardGlassHeader, CardGlassRoot, CardGlassTitle, CardHeader, CardTitle, CareerStatsGlass, CareerStatsHeaderGlass, Cell, ChartContainer, ChartLegend, ChartLegendContent, ChartStyle, ChartTooltip, ChartTooltipContent, Checkbox, CheckboxGlass, CircularMetricGlass, CircularProgressGlass, ComboBoxGlass, ContributionMetricsGlass, Dialog, DialogClose, DialogContent, DialogDescription, DialogFooter, DialogHeader, DialogOverlay, DialogPortal, DialogTitle, DialogTrigger, DropdownGlass, DropdownMenu, DropdownMenuCheckboxItem, DropdownMenuContent, DropdownMenuGlass, DropdownMenuGlassCheckboxItem, DropdownMenuGlassContent, DropdownMenuGlassGroup, DropdownMenuGlassItem, DropdownMenuGlassLabel, DropdownMenuGlassPortal, DropdownMenuGlassRadioGroup, DropdownMenuGlassRadioItem, DropdownMenuGlassSeparator, DropdownMenuGlassShortcut, DropdownMenuGlassSub, DropdownMenuGlassSubContent, DropdownMenuGlassSubTrigger, DropdownMenuGlassTrigger, DropdownMenuGroup, DropdownMenuItem, DropdownMenuLabel, DropdownMenuPortal, DropdownMenuRadioGroup, DropdownMenuRadioItem, DropdownMenuSeparator, DropdownMenuShortcut, DropdownMenuSub, DropdownMenuSubContent, DropdownMenuSubTrigger, DropdownMenuTrigger, ExpandableHeaderGlass, FlagAlertGlass, FlagsSectionGlass, FormFieldWrapper, GlassCard, HeaderBrandingGlass, HeaderNavGlass, HoverCardGlassRoot as HoverCard, HoverCardGlassContent as HoverCardContent, HoverCardGlassContent, HoverCardGlass, HoverCardGlassLegacy, HoverCardGlassTrigger, HoverCardGlassTrigger as HoverCardTrigger, IconButtonGlass, Input, InputGlass, InsightCardGlass, InteractiveCard, LanguageBarGlass, MetricCardGlass, MetricsGridGlass, ModalBody, ModalClose, ModalContent, ModalDescription, ModalFooter, ModalGlass, ModalHeader, ModalOverlay, ModalPortal, ModalRoot, ModalTitle, ModalTrigger, NotificationGlass, PopoverGlass, PopoverGlassAnchor, PopoverGlassContent, PopoverGlassLegacy, PopoverGlassTrigger, ProfileAvatarGlass, ProfileHeaderExtendedGlass, ProfileHeaderGlass, Progress, ProgressGlass, ProjectsListGlass, RainbowProgressGlass, RepositoryCardActions, RepositoryCardExpandedContent, RepositoryCardGlass, RepositoryCardHeader, RepositoryCardIssues, RepositoryCardLanguages, RepositoryCardMeta, RepositoryCardMetricItem, RepositoryCardMetrics, RepositoryCardName, RepositoryCardRoot, RepositoryCardStars, RepositoryCardStatsLine, RepositoryCardStatus, RepositoryHeaderGlass, RepositoryMetadataGlass, ResponsiveContainer, ScrollAreaGlass as ScrollArea, ScrollAreaGlass, ScrollAreaGlassViewport, ScrollBarGlass as ScrollBar, ScrollBarGlass, SearchBoxGlass, SegmentedControlGlass, Select, SelectContent, SelectGlass, SelectGlassContent, SelectGlassGroup, SelectGlassItem, SelectGlassLabel, SelectGlassScrollDownButton, SelectGlassScrollUpButton, SelectGlassSeparator, SelectGlassTrigger, SelectGlassValue, SelectGroup, SelectItem, SelectLabel, SelectScrollDownButton, SelectScrollUpButton, SelectSeparator, SelectTrigger, SelectValue, SeparatorGlass as Separator, SeparatorGlass, Sheet, SheetClose, SheetContent, SheetDescription, SheetFooter, SheetGlass, SheetHeader, SheetOverlay, SheetPortal, SheetRoot, SheetTitle, SheetTrigger, SidebarGlass, Skeleton, SkeletonGlass, SliderGlass, SortDropdownGlass, SparklineGlass, SplitLayoutAccordion, SplitLayoutGlass, StatItemGlass, StatusIndicatorGlass, StepperGlass, Switch, SwitchGlass, THEMES, THEME_CONFIG, Tabs, TabsContent, TabsGlass, TabsList, TabsTrigger, ThemeProvider, ThemeToggleGlass, Toggle, ToggleGlass, Tooltip, TooltipContent, TooltipGlass, TooltipGlassContent, TooltipGlassProvider, TooltipGlassSimple, TooltipGlassTrigger, TooltipProvider, TooltipTrigger, TouchTarget, TrustScoreCardGlass, TrustScoreDisplayGlass, UserInfoGlass, UserStatsLineGlass, YearCardAction, YearCardBadge, YearCardExpandedContent, YearCardGlass, YearCardHeader, YearCardInsightItem, YearCardInsights, YearCardProgress, YearCardRoot, YearCardSparkline, YearCardStatItem, YearCardStats, YearCardValue, YearCardYear, alertVariants, avatarSizes, badgeVariants, buttonGlassVariants, buttonGlassVariants as buttonVariants, cardIntensity, cn, dropdownAlign, getNextTheme, getThemeConfig, inputVariants, insightCardVariants, insightVariantConfig, modalSizes, notificationVariants, progressSizes, alertVariants$1 as shadcnAlertVariants, badgeVariants$1 as shadcnBadgeVariants, buttonVariants as shadcnButtonVariants, sheetVariants, skeletonVariants, sparklineBarVariants, sparklineContainerVariants, statusSizes, stepperConnectorVariants, stepperContentVariants, stepperDescriptionVariants, stepperIndicatorVariants, stepperLabelVariants, stepperListVariants, stepperRootVariants, stepperStepContainerVariants, toggleSizes, tooltipPositions, useChart, useFocus, useHover, useRepositoryCard, useRepositoryCardOptional, useResponsive, useSidebar, useSplitLayout, useSplitLayoutOptional, useTheme, useWallpaperTint, useYearCard, useYearCardOptional };
