// shadcn-glass-ui/dist/cli/index.mjs
#!/usr/bin/env node
import { parseArgs } from "node:util";
import { readFileSync } from "node:fs";
import { dirname, resolve } from "node:path";
import { fileURLToPath } from "node:url";
var cachedExportsMap = null;
function loadExportsMap() {
	if (cachedExportsMap) return cachedExportsMap;
	const __dirname = dirname(fileURLToPath(import.meta.url));
	const paths = [
		resolve(__dirname, "../../../docs/EXPORTS_MAP.json"),
		resolve(__dirname, "../../../docs/EXPORTS_MAP.json"),
		resolve(process.cwd(), "node_modules/shadcn-glass-ui/docs/EXPORTS_MAP.json"),
		resolve(process.cwd(), "docs/EXPORTS_MAP.json")
	];
	for (const path of paths) try {
		const content = readFileSync(path, "utf-8");
		cachedExportsMap = JSON.parse(content);
		return cachedExportsMap;
	} catch {
		continue;
	}
	throw new Error("EXPORTS_MAP.json not found. Please ensure:\n  - You are running from the project root, or\n  - The shadcn-glass-ui package is installed correctly");
}
function getAllComponents() {
	const exportsMap = loadExportsMap();
	const components = [];
	for (const [categoryKey, categoryData] of Object.entries(exportsMap.components)) if (categoryData.exports) for (const exp of categoryData.exports) components.push({
		name: exp.name,
		path: exp.path,
		category: categoryKey,
		level: categoryData.level ?? 0
	});
	return components;
}
function findComponentByName(name) {
	const exportsMap = loadExportsMap();
	for (const [categoryKey, categoryData] of Object.entries(exportsMap.components)) if (categoryData.exports) {
		const found = categoryData.exports.find((exp) => exp.name === name);
		if (found) return {
			component: found,
			category: categoryKey,
			level: categoryData.level ?? 0
		};
	}
	return null;
}
function getCategories() {
	const exportsMap = loadExportsMap();
	const categories = [];
	for (const [key, data] of Object.entries(exportsMap.components)) categories.push({
		key,
		level: data.level ?? 0,
		description: data.description,
		count: data.exports?.length ?? 0
	});
	return categories.sort((a, b) => a.level - b.level);
}
function getStatistics() {
	const exportsMap = loadExportsMap();
	return {
		totalComponents: exportsMap.statistics.totalComponents,
		byLevel: exportsMap.statistics.byLevel,
		version: exportsMap.version
	};
}
function normalizeQuery(query) {
	return query.toLowerCase().replace(/glass$/i, "").replace(/-/g, "").replace(/_/g, "").trim();
}
function normalizeComponentName(name) {
	return name.toLowerCase().replace(/glass$/i, "").replace(/-/g, "").replace(/_/g, "");
}
function levenshteinDistance(a, b) {
	const matrix = [];
	for (let i = 0; i <= b.length; i++) matrix[i] = [i];
	for (let j = 0; j <= a.length; j++) matrix[0][j] = j;
	for (let i = 1; i <= b.length; i++) for (let j = 1; j <= a.length; j++) if (b.charAt(i - 1) === a.charAt(j - 1)) matrix[i][j] = matrix[i - 1][j - 1];
	else matrix[i][j] = Math.min(matrix[i - 1][j - 1] + 1, matrix[i][j - 1] + 1, matrix[i - 1][j] + 1);
	return matrix[b.length][a.length];
}
function calculateMatchScore(query, componentName) {
	const normalizedQuery = normalizeQuery(query);
	const normalizedName = normalizeComponentName(componentName);
	if (normalizedName === normalizedQuery) return 100;
	if (componentName.toLowerCase() === query.toLowerCase()) return 100;
	if (normalizedName.startsWith(normalizedQuery)) return 80;
	if (normalizedName.includes(normalizedQuery)) return 60;
	const distance = levenshteinDistance(normalizedQuery, normalizedName);
	if (distance <= (normalizedQuery.length <= 4 ? 2 : 3)) return Math.max(20, 50 - distance * 10);
	return 0;
}
function fuzzyMatch(query, components) {
	if (!query || query.trim().length === 0) return [];
	return components.map((comp) => ({
		...comp,
		score: calculateMatchScore(query, comp.name)
	})).filter((match) => match.score > 0).sort((a, b) => {
		if (b.score !== a.score) return b.score - a.score;
		if (a.level !== b.level) return a.level - b.level;
		return a.name.localeCompare(b.name);
	});
}
const COLORS = {
	reset: "\x1B[0m",
	bold: "\x1B[1m",
	dim: "\x1B[2m",
	italic: "\x1B[3m",
	underline: "\x1B[4m",
	black: "\x1B[30m",
	red: "\x1B[31m",
	green: "\x1B[32m",
	yellow: "\x1B[33m",
	blue: "\x1B[34m",
	magenta: "\x1B[35m",
	cyan: "\x1B[36m",
	white: "\x1B[37m",
	brightRed: "\x1B[91m",
	brightGreen: "\x1B[92m",
	brightYellow: "\x1B[93m",
	brightBlue: "\x1B[94m",
	brightMagenta: "\x1B[95m",
	brightCyan: "\x1B[96m"
};
const CATEGORY_EMOJI = {
	primitives: "🧱",
	core: "⚙️",
	atomic: "🔹",
	specialized: "🎯",
	composite: "🧩",
	sections: "📄"
};
function color(text, colorCode) {
	return `${colorCode}${text}${COLORS.reset}`;
}
function bold(text) {
	return `${COLORS.bold}${text}${COLORS.reset}`;
}
function dim(text) {
	return `${COLORS.dim}${text}${COLORS.reset}`;
}
function line(length = 50) {
	return dim("─".repeat(length));
}
function success(text) {
	return `${COLORS.green}✓${COLORS.reset} ${text}`;
}
function bullet(text, colorCode = COLORS.cyan) {
	return `  ${colorCode}•${COLORS.reset} ${text}`;
}
function sectionHeader(title) {
	return `\n${COLORS.bold}${title}:${COLORS.reset}`;
}
function keyValue(key, value, keyWidth = 12) {
	const paddedKey = key.padEnd(keyWidth);
	return `${COLORS.bold}${paddedKey}${COLORS.reset} ${value}`;
}
function generateUsageExample(comp) {
	const name = comp.name;
	if (comp.compoundAPI && comp.subComponents && comp.subComponents.length > 0) {
		const subComps = comp.subComponents;
		const contentComp = subComps.find((s) => s.includes("Content")) || subComps[1] || subComps[0];
		return `<${name}.Root>
  <${name}.${contentComp}>
    {/* Content here */}
  </${name}.${contentComp}>
</${name}.Root>`;
	}
	const props = [];
	if (comp.variants && comp.variants.length > 0) props.push(`variant="${comp.variants[0]}"`);
	if (comp.sizes && comp.sizes.length > 0) props.push(`size="${comp.sizes[0]}"`);
	const propsStr = props.length > 0 ? " " + props.join(" ") : "";
	if ([
		"Avatar",
		"Progress",
		"Skeleton",
		"Slider",
		"Checkbox",
		"Toggle",
		"Input"
	].some((type) => name.includes(type))) return `<${name}${propsStr} />`;
	return `<${name}${propsStr}>
  {/* Content here */}
</${name}>`;
}
function printComponentInfo(comp, category, level, alternativeMatches) {
	console.log(`\n${color(bold(comp.name), COLORS.cyan)}`);
	console.log(line(50));
	if (alternativeMatches.length > 1) {
		console.log(dim(`Found ${alternativeMatches.length} matches. Showing best match.`));
		if (alternativeMatches.length <= 5) {
			const others = alternativeMatches.slice(1).map((m) => m.name).join(", ");
			console.log(dim(`Other matches: ${others}`));
		}
		console.log("");
	}
	console.log(sectionHeader("Description"));
	console.log(`  ${comp.description}`);
	console.log("");
	console.log(keyValue("Category:", category));
	console.log(keyValue("Level:", `L${level}`));
	console.log(keyValue("Type:", comp.type));
	console.log(keyValue("Path:", comp.path));
	if (comp.props) console.log(keyValue("Props:", comp.props));
	if (comp.variants && comp.variants.length > 0) {
		console.log(sectionHeader("Variants"));
		comp.variants.forEach((v) => {
			console.log(bullet(v, COLORS.green));
		});
	}
	if (comp.sizes && comp.sizes.length > 0) {
		console.log(sectionHeader("Sizes"));
		comp.sizes.forEach((s) => {
			console.log(bullet(s, COLORS.yellow));
		});
	}
	if (comp.statuses && comp.statuses.length > 0) {
		console.log(sectionHeader("Statuses"));
		comp.statuses.forEach((s) => {
			console.log(bullet(s, COLORS.blue));
		});
	}
	console.log("");
	if (comp.shadcnCompatible) console.log(success("shadcn/ui compatible"));
	if (comp.compoundAPI) {
		console.log(success("Compound component API"));
		if (comp.subComponents && comp.subComponents.length > 0) console.log(dim(`  Sub-components: ${comp.subComponents.join(", ")}`));
	}
	if (comp.supportsAsChild) console.log(success("Supports asChild pattern"));
	console.log(sectionHeader("Usage Example"));
	console.log(line(21));
	console.log("");
	console.log(color(`import { ${comp.name} } from 'shadcn-glass-ui';`, COLORS.cyan));
	console.log("");
	console.log(generateUsageExample(comp));
	if (comp.keywords && comp.keywords.length > 0) {
		console.log("");
		console.log(dim(`Keywords: ${comp.keywords.join(", ")}`));
	}
	console.log("");
}
async function showComponentInfo(query) {
	const matches = fuzzyMatch(query, getAllComponents());
	if (matches.length === 0) {
		console.log(`\n${color("No component found matching", COLORS.red)} "${query}"`);
		console.log(dim("\nTry one of these:"));
		console.log(dim("  npx shadcn-glass-ui list"));
		console.log(dim("  npx shadcn-glass-ui info button"));
		console.log(dim("  npx shadcn-glass-ui info modal"));
		console.log("");
		process.exit(1);
	}
	const bestMatch = matches[0];
	const details = findComponentByName(bestMatch.name);
	if (!details) {
		console.log(`\n${color("Component details not found for", COLORS.red)} "${bestMatch.name}"`);
		process.exit(1);
	}
	printComponentInfo(details.component, details.category, details.level, matches);
}
function printComponent(comp) {
	const flags = [];
	if (comp.shadcnCompatible) flags.push(color("shadcn", COLORS.green));
	if (comp.compoundAPI) flags.push(color("compound", COLORS.yellow));
	if (comp.supportsAsChild) flags.push(color("asChild", COLORS.magenta));
	const flagStr = flags.length > 0 ? ` [${flags.join(", ")}]` : "";
	console.log(`   ${color("•", COLORS.cyan)} ${comp.name}${flagStr}`);
}
function printCategory(key, level, description, components) {
	const emoji = CATEGORY_EMOJI[key] || "📦";
	const levelStr = `L${level}`;
	console.log(`${emoji} ${bold(key.toUpperCase())} ${dim(`(${levelStr})`)}`);
	console.log(`   ${dim(description)}`);
	console.log("");
	for (const comp of components) printComponent(comp);
	console.log("");
}
async function listComponents(options = {}) {
	const exportsMap = loadExportsMap();
	const categories = getCategories();
	const stats = getStatistics();
	console.log(`\n${color(bold("shadcn-glass-ui Components"), COLORS.cyan)}`);
	console.log(`${dim(`Version ${stats.version}`)}`);
	console.log(line(50));
	console.log("");
	let totalCount = 0;
	let filteredCount = 0;
	const categoryFilter = options.category?.toLowerCase();
	const levelFilter = options.level !== void 0 ? parseInt(options.level, 10) : void 0;
	for (const cat of categories) {
		const categoryData = exportsMap.components[cat.key];
		if (!categoryData || !categoryData.exports || categoryData.exports.length === 0) continue;
		if (categoryFilter && cat.key.toLowerCase() !== categoryFilter) {
			totalCount += categoryData.exports.length;
			continue;
		}
		if (levelFilter !== void 0 && cat.level !== levelFilter) {
			totalCount += categoryData.exports.length;
			continue;
		}
		printCategory(cat.key, cat.level, cat.description, categoryData.exports);
		totalCount += categoryData.exports.length;
		filteredCount += categoryData.exports.length;
	}
	console.log(line(21));
	if (categoryFilter || levelFilter !== void 0) console.log(`${bold(`Showing: ${filteredCount}`)} ${dim(`of ${totalCount} total components`)}`);
	else console.log(`${bold(`Total: ${totalCount} components`)}`);
	console.log("");
	if (!categoryFilter && levelFilter === void 0) {
		console.log(dim("Filter by category: npx shadcn-glass-ui list --category=core"));
		console.log(dim("Filter by level:    npx shadcn-glass-ui list --level=1"));
	}
	console.log(dim("Get component info: npx shadcn-glass-ui info <name>"));
	console.log("");
}
function printAvailableCategories() {
	const categories = getCategories();
	console.log("\nAvailable categories:");
	for (const cat of categories) {
		const emoji = CATEGORY_EMOJI[cat.key] || "📦";
		console.log(`  ${emoji} ${cat.key} (L${cat.level}) - ${cat.count} components`);
	}
	console.log("");
}
function getVersion() {
	const __dirname = dirname(fileURLToPath(import.meta.url));
	const paths = [
		resolve(__dirname, "../../package.json"),
		resolve(__dirname, "../../../package.json"),
		resolve(process.cwd(), "package.json"),
		resolve(process.cwd(), "node_modules/shadcn-glass-ui/package.json")
	];
	for (const path of paths) try {
		const pkg = JSON.parse(readFileSync(path, "utf-8"));
		if (pkg.name === "shadcn-glass-ui") return pkg.version;
	} catch {
		continue;
	}
	return "unknown";
}
function printHelp() {
	console.log(`
${color(bold("shadcn-glass-ui"), COLORS.cyan)} - Glassmorphism UI Component Library

${bold("Usage:")}
  npx shadcn-glass-ui <command> [options]

${bold("Commands:")}
  info <name>    Show detailed component information
  list           List all available components

${bold("Options:")}
  --help, -h     Show this help message
  --version, -v  Show version number
  --category     Filter list by category (e.g., --category=core)
  --level        Filter list by level (e.g., --level=1)

${bold("Examples:")}
  ${dim("# Get info about a component (fuzzy search supported)")}
  npx shadcn-glass-ui info ButtonGlass
  npx shadcn-glass-ui info button
  npx shadcn-glass-ui info modal

  ${dim("# List all components")}
  npx shadcn-glass-ui list

  ${dim("# List components by category")}
  npx shadcn-glass-ui list --category=core
  npx shadcn-glass-ui list --category=composite

  ${dim("# List components by level")}
  npx shadcn-glass-ui list --level=1

${bold("Categories:")}
  primitives  (L0)  Foundation building blocks
  core        (L1)  Basic interactive components
  atomic      (L2)  Small specialized components
  specialized (L3)  Domain-specific components
  composite   (L4)  Multi-element widgets
  sections    (L5)  Full page sections

${bold("More info:")}
  Documentation: https://yhooi2.github.io/shadcn-glass-ui-library/
  Repository:    https://github.com/Yhooi2/shadcn-glass-ui-library
`);
}
function parseCliArgs() {
	const config = {
		allowPositionals: true,
		strict: false,
		options: {
			help: {
				type: "boolean",
				short: "h",
				default: false
			},
			version: {
				type: "boolean",
				short: "v",
				default: false
			},
			category: { type: "string" },
			level: { type: "string" }
		}
	};
	try {
		const { values, positionals } = parseArgs(config);
		const [command, ...args] = positionals;
		return {
			command,
			args,
			options: {
				help: values.help,
				version: values.version,
				category: values.category,
				level: values.level
			}
		};
	} catch (error) {
		if (error instanceof Error && error.message.includes("Unknown option")) {
			console.error(`${color("Error:", COLORS.red)} ${error.message}`);
			console.log(dim("\nRun \"npx shadcn-glass-ui --help\" for usage information."));
			process.exit(1);
		}
		throw error;
	}
}
async function main() {
	const { command, args, options } = parseCliArgs();
	if (options.version) {
		console.log(getVersion());
		process.exit(0);
	}
	if (options.help || !command) {
		printHelp();
		process.exit(0);
	}
	switch (command.toLowerCase()) {
		case "info":
			if (args.length === 0) {
				console.error(`${color("Error:", COLORS.red)} Component name required`);
				console.log(dim("\nUsage: npx shadcn-glass-ui info <ComponentName>"));
				console.log(dim("Example: npx shadcn-glass-ui info ButtonGlass"));
				process.exit(1);
			}
			await showComponentInfo(args[0]);
			break;
		case "list":
			await listComponents({
				category: options.category,
				level: options.level
			});
			break;
		case "categories":
			printAvailableCategories();
			break;
		default:
			console.error(`${color("Error:", COLORS.red)} Unknown command "${command}"`);
			console.log(dim("\nAvailable commands: info, list"));
			console.log(dim("Run \"npx shadcn-glass-ui --help\" for usage information."));
			process.exit(1);
	}
}
main().catch((error) => {
	console.error(`${color("Error:", COLORS.red)} ${error.message}`);
	if (process.env.DEBUG) console.error(error.stack);
	process.exit(1);
});
